{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Assignment 2. Intensive integration using Web APIs\n",
    "## Gema Castillo GarcÃ­a"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Files I have created:\n",
    "Scripts defining each Class:\n",
    "\n",
    "    Interaction_Network.rb\n",
    "    Uso_General_Annotation.rb\n",
    "\n",
    "**The 2 Classes are interconnected. The values of some Object Properties are other Objects: the networks are the key to link Interaction_Network > Uso_General_Annotation.** \n",
    "\n",
    "\n",
    "The main script that uses the previous 2 scripts to make the report is:\n",
    "\n",
    "    make_report.rb\n",
    "\n",
    "\n",
    "### Final program\n",
    "    $ ruby make_report.rb\n",
    "    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       ":fetch"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "require 'rest-client'  \n",
    "\n",
    "#creating a function called \"fetch\" that we can re-use everywhere in our code ()\n",
    "def fetch(url, headers = {accept: \"*/*\"}, user = \"\", pass=\"\") #the definition of fetch was written by Mark Wilkinson\n",
    "  response = RestClient::Request.execute({\n",
    "    method: :get,\n",
    "    url: url.to_s,\n",
    "    user: user,\n",
    "    password: pass,\n",
    "    headers: headers})\n",
    "  return response\n",
    "  \n",
    "  rescue RestClient::ExceptionWithResponse => e\n",
    "    $stderr.puts e.inspect\n",
    "    response = false\n",
    "    return response  #now we are returning 'False', and we will check that with an \\\"if\\\" statement in our main code\n",
    "  rescue RestClient::Exception => e\n",
    "    $stderr.puts e.inspect\n",
    "    response = false\n",
    "    return response  #now we are returning 'False', and we will check that with an \\\"if\\\" statement in our main code\n",
    "  rescue Exception => e\n",
    "    $stderr.puts e.inspect\n",
    "    response = false\n",
    "    return response  #now we are returning 'False', and we will check that with an \\\"if\\\" statement in our main code\n",
    "end"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       ":show_all_networks"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#the code of this box is in Interaction_Network.rb file\n",
    "#'ArabidopsisSubNetwork_GeneList.txt' contains 168 genes\n",
    "\n",
    "#beginning a class definition to show the Interaction Network\n",
    "class Interaction_Network\n",
    "  #creating \"attribute accessors\" to read and write objects' properties\n",
    "  attr_accessor :agi_locus1\n",
    "  attr_accessor :agi_locus2\n",
    "  attr_accessor :interactors\n",
    "  \n",
    "  #initialization inside the class\n",
    "  def initialize(agi_locus1 ,agi_locus2 ,interactors ) \n",
    "    @agi_locus1 = agi_locus1 #this attribute is for the genes of the list that I have used to do the search in BAR database\n",
    "    @agi_locus2 = agi_locus2 #this attribute is for the genes of the list that have some interactors in common with agi_locus1\n",
    "    @interactors = interactors #this attribute is for the interactors of each gene that are not present in the list\n",
    "  end\n",
    "  \n",
    "  \n",
    "  def Interaction_Network.retrieve_networks #defining a method to store the interactors of each gene in an array called 'gene_interactors'\n",
    "    puts \"\\nSearching for interactions in BAR database from UToronto...\\n\"\n",
    "    agi = [] #creating a new array that will contain the genes' AGI of \"ArabidopsisSubNetwork_GeneList.txt\"\n",
    "    @@networks = [] #creating an array to store the Interaction_Network Objects\n",
    "    agi = File.read(\"ArabidopsisSubNetwork_GeneList.txt\").split #retrieving the genes' AGI of \"ArabidopsisSubNetwork_GeneList.txt\"\n",
    "    taxid = 'taxid:3702' #taxon ID for A.thaliana\n",
    "    cutoff = 0.485 #MIscore cutoff value for optimal score predictions found by calcuating the maximal Matthews correlation coefficient (MCC) in https://europepmc.org/article/MED/25652942\n",
    "    for $g in 0..agi.length-1 #for each gene of the list do...\n",
    "      agi[$g][\"T\"]= \"t\" #changing 'T' for 't' to follow the BAR database nomenclature \n",
    "      gene_interactors = [] #creating an array for the AGI code of each interactor\n",
    "      res = fetch(\"http://bar.utoronto.ca:9090/psicquic/webservices/current/search/query/#{agi[$g]}?format=tab25\") #accessing genes URLs\n",
    "      if res  #res is either the response object, or False, so you can test it with 'if'\n",
    "        body = res.body  #getting the \"body\" of the response\n",
    "        if body.empty?\n",
    "          puts \"There is no entry for #{agi[$g]} in BAR database from UToronto.\" #printing a friendly message if the gene has not any interactor\n",
    "        else #formatting the URL content to retrieve the required information\n",
    "          lines = res.body.split(\"\\n\")\n",
    "          lines.each do |l|\n",
    "            fields = l.split(\"\\t\")\n",
    "            if fields[9].include? taxid and fields[10].include? taxid and fields[14][15..19].to_f > cutoff #selecting A.thaliana genes and interactions with optimal quality\n",
    "              if fields[0] != fields[1] #not including the interactor if the protein interacts with itself\n",
    "                if fields[2] =~ /tair:([^\"]+)/ #searching for reg expressions in the first column of interactors\n",
    "                  if $1 != agi[$g] #selecting only the interactors, NOT the query\n",
    "                    gene_interactors << $1\n",
    "                  end\n",
    "                end\n",
    "                if fields[3] =~ /tair:([^\"]+)/ #searching for reg expressions in the second column of interactors\n",
    "                  if $1 != agi[$g]  #selecting only the interactors, NOT the query\n",
    "                    gene_interactors << $1\n",
    "                  end\n",
    "                end\n",
    "              end\n",
    "            end\n",
    "          end\n",
    "        end\n",
    "      end\n",
    "      gene_interactors = gene_interactors.uniq #removing repeated interactors\n",
    "      #gene_interactors = agi & gene_interactors ##to remove AGI not present in \"ArabidopsisSubNetwork_GeneList.txt\"\n",
    "                                                 ##(it is hidden because I am also considering the interactors that are not present in the list)  \n",
    "      for $i in 0..gene_interactors.length-1\n",
    "        #puts \"#{agi[$g]} interacts with #{gene_interactors[$i]}\"\n",
    "        @@networks << Interaction_Network.new(agi[$g],\"none\",gene_interactors) #creating an array with the new Objects\n",
    "      end\n",
    "    end\n",
    "  end\n",
    "  \n",
    "  \n",
    "  def Interaction_Network.show_networks #defining a method to work with Interaction_Network Class objects\n",
    "    return @@networks\n",
    "  end\n",
    "  \n",
    "  \n",
    "  def Interaction_Network.binding_networks #defining a method to connect different objects to make bigger networks\n",
    "    puts \"\\nBinding networks...\\n\"\n",
    "    @@all_networks = [] #creating an array to store the new Interaction_Network Objects\n",
    "    common_interactors = [] #creating an array to store the genes of the list that interact with each other\n",
    "    Interaction_Network.show_networks.each do |n1|\n",
    "      Interaction_Network.show_networks.each do |n2|\n",
    "        unless n1.agi_locus1 == n2.agi_locus1 #ignoring comparisons from a gene with itself\n",
    "          unless n1.interactors & n2.interactors == [] #dismissing the genes of the list without interactors in common\n",
    "            if n2.interactors.include? n1.agi_locus1 #searching direct interactions between the genes of the list\n",
    "              common_interactors << [n1.agi_locus1, n2.agi_locus1, [n1.interactors & n2.interactors, \"direct\"]]\n",
    "            else #searching indirect interactions between the genes of the list\n",
    "              common_interactors << [n1.agi_locus1, n2.agi_locus1, n1.interactors & n2.interactors]\n",
    "            end\n",
    "          else #searching direct interactions between the genes of the list without interactors in common\n",
    "            if n2.interactors.include? n1.agi_locus1\n",
    "              common_interactors << [n1.agi_locus1, n2.agi_locus1, \"direct\"]\n",
    "            end   \n",
    "          end\n",
    "        end \n",
    "  \n",
    "      end\n",
    "    end\n",
    "\n",
    "    common_interactors = common_interactors.uniq #deleting the duplicated networks\n",
    "    common_interactors2 = [] #creating an array to store definitive networks without duplicates\n",
    "    common_interactors.each do |c|\n",
    "      common_interactors2 << c.sort_by do |cc| #sorting by alphabet the members of each network to compare different networks and delete the duplicates\n",
    "        cc.class == Array ? cc.to_s : cc\n",
    "      end\n",
    "    end\n",
    "    \n",
    "    common_interactors2 = common_interactors2.uniq\n",
    "    common_interactors2.each do |ci|\n",
    "      @@all_networks << Interaction_Network.new(ci[0],ci[1],ci[2]) #creating an array with the new Objects with bigger networks\n",
    "    end\n",
    "  end\n",
    "  \n",
    "  \n",
    "  def Interaction_Network.show_all_networks #defining a method to work with Interaction_Network Class networks\n",
    "    return @@all_networks\n",
    "  end\n",
    "  \n",
    "  \n",
    "end"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       ":show_annotated_objects"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#the code of this box is in Uso_General_Annotation.rb file \n",
    "##BONUS: this Class can hold any functional annotation as long as you create the corresponding attribute and function\n",
    "\n",
    "require 'json'\n",
    "\n",
    "class Uso_General_Annotation < Interaction_Network #Uso_General_Annotation is inheriting the properties of Interaction_Network\n",
    "  #creating \"attribute accessors\" to read and write objects' properties\n",
    "  attr_accessor :network_members \n",
    "  attr_accessor :kegg \n",
    "  attr_accessor :go \n",
    "  \n",
    "  #initialization inside the class\n",
    "  def initialize(network_members ,kegg ,go ) \n",
    "    @network_members = network_members #this attribute will contain the objects with the networks created in Interaction_Network Class\n",
    "    @kegg = kegg #this attribute will contain the KEGG pathways of all members of each network\n",
    "    @go = go #this attribute will contain the GO terms of all members of each network\n",
    "  end\n",
    "  \n",
    "  \n",
    "  def Uso_General_Annotation.GO_annotation\n",
    "    puts \"\\nRetrieving GO terms...\\n\"\n",
    "    goterms = [] #creating an array for the GO terms of each interactor\n",
    "    @@allgoterms = [] #creating an array for the GO terms of each network\n",
    "    Interaction_Network.show_all_networks.each do |net|\n",
    "      res = fetch(\"https://www.ebi.ac.uk/Tools/dbfetch/dbfetch?db=uniprotkb&id=#{net.agi_locus1}&style=raw\") #accessing the URL of the first gene of the networks\n",
    "      if res#res is either the response object, or False, so you can test it with 'if'\n",
    "        body = res.body #getting the \"body\" of the response\n",
    "        if body.empty?\n",
    "          goterms = []\n",
    "        else\n",
    "          lines = res.body.split(\"\\n\")\n",
    "          lines.each do |l|\n",
    "            if l =~ /GO:([^\"]+); P:([^\"]+);/ #selecting the GO terms from the biological process part of the GO Ontology\n",
    "              goterms << [\"#{$1}\", \"#{$2}\"] #storing the GO IDs and Terms, respectively\n",
    "              goterms = goterms.uniq #deleting duplicated GO terms present in the same URL\n",
    "            end\n",
    "          end\n",
    "        end\n",
    "      end\n",
    "      \n",
    "      res = fetch(\"https://www.ebi.ac.uk/Tools/dbfetch/dbfetch?db=uniprotkb&id=#{net.agi_locus2}&style=raw\") #accessing the URL of the second gene of the networks\n",
    "      if res #res is either the response object, or False, so you can test it with 'if'\n",
    "        body = res.body #getting the \"body\" of the response\n",
    "        if body.empty?\n",
    "          goterms = []\n",
    "        else\n",
    "          lines = res.body.split(\"\\n\")\n",
    "          lines.each do |l|\n",
    "            if l =~ /GO:([^\"]+); P:([^\"]+);/ #selecting the GO terms from the biological process part of the GO Ontology\n",
    "              goterms << [\"#{$1}\", \"#{$2}\"] #storing the GO IDs and Terms, respectively\n",
    "              goterms = goterms.uniq #deleting duplicated GO terms present in the same URL\n",
    "            end\n",
    "          end\n",
    "        end\n",
    "      end\n",
    "      \n",
    "      #this block is very long because it has to access all the interactors (they are in a different format)\n",
    "      if net.interactors.class == Array #accessing the URLs of the interactors in common of the networks\n",
    "        net.interactors.each do |int|\n",
    "          unless int.class == String\n",
    "            int.each do |intt|\n",
    "              res = fetch(\"https://www.ebi.ac.uk/Tools/dbfetch/dbfetch?db=uniprotkb&id=#{intt}&style=raw\") #accessing interactor's URLs\n",
    "              if res #res is either the response object, or False, so you can test it with 'if'\n",
    "                body = res.body #getting the \"body\" of the response\n",
    "                if body.empty?\n",
    "                  goterms = []\n",
    "                else\n",
    "                  lines = res.body.split(\"\\n\")\n",
    "                  lines.each do |l|\n",
    "                    if l =~ /GO:([^\"]+); P:([^\"]+);/ #selecting the GO terms from the biological process part of the GO Ontology\n",
    "                      goterms << [\"#{$1}\", \"#{$2}\"] #storing the GO IDs and Terms, respectively\n",
    "                      goterms = goterms.uniq #deleting duplicated GO terms present in the same URL\n",
    "                    end\n",
    "                  end\n",
    "                end\n",
    "              end\n",
    "            end\n",
    "          else\n",
    "            unless int == \"direct\"\n",
    "              res = fetch(\"https://www.ebi.ac.uk/Tools/dbfetch/dbfetch?db=uniprotkb&id=#{int}&style=raw\") #accessing interactor's URLs\n",
    "              if res#res is either the response object, or False, so you can test it with 'if'\n",
    "                body = res.body #getting the \"body\" of the response\n",
    "                if body.empty?\n",
    "                  goterms = []\n",
    "                else\n",
    "                  lines = res.body.split(\"\\n\")\n",
    "                  lines.each do |l|\n",
    "                    if l =~ /GO:([^\"]+); P:([^\"]+);/ #selecting the GO terms from the biological process part of the GO Ontology\n",
    "                      goterms << [\"#{$1}\", \"#{$2}\"] #storing the GO IDs and Terms, respectively\n",
    "                      goterms = goterms.uniq #deleting duplicated GO terms present in the same URL\n",
    "                    end\n",
    "                  end\n",
    "                end\n",
    "              end\n",
    "            end\n",
    "          end\n",
    "        end\n",
    "      end\n",
    "      @@allgoterms << goterms #storing the GO terms of each network in an array\n",
    "      goterms = []\n",
    "    end\n",
    "  end\n",
    "  \n",
    "  \n",
    "  def Interaction_Network.show_go #defining a method to work with GO annotations\n",
    "    return @@allgoterms\n",
    "  end\n",
    "  \n",
    "  \n",
    "  def Uso_General_Annotation.KEGG_annotation\n",
    "    puts \"\\nRetrieving KEGG pathways...\\n\"\n",
    "    kegg = [] #creating an array for the KEGG pathways of each interactor\n",
    "    @@allkegg = [] #creating an array for the KEGG pathways of each network\n",
    "    Interaction_Network.show_all_networks.each do |net|\n",
    "      address = \"http://togows.org/entry/kegg-genes/ath:#{net.agi_locus1}/pathways.json\" #accessing the URL of the first gene of the networks\n",
    "      response = RestClient::Request.execute(\n",
    "        method: :get,\n",
    "        url: address)\n",
    "      kegg_data = JSON.parse(response.body)\n",
    "      for elem in kegg_data[0].each\n",
    "        if kegg_data[0].empty?\n",
    "          kegg = []\n",
    "        elsif kegg_data[0]\n",
    "          kegg << [elem[0],elem[1]] #storing the KEGG IDs and Pathways, respectively\n",
    "        end\n",
    "      end\n",
    "      \n",
    "      address = \"http://togows.org/entry/kegg-genes/ath:#{net.agi_locus2}/pathways.json\" #accessing the URL of the second gene of the networks\n",
    "      response = RestClient::Request.execute(\n",
    "        method: :get,\n",
    "        url: address)\n",
    "      kegg_data = JSON.parse(response.body)\n",
    "      for elem in kegg_data[0].each\n",
    "        if kegg_data[0].empty?\n",
    "          kegg = []\n",
    "        elsif kegg_data[0]\n",
    "          kegg << [elem[0],elem[1]] #storing the KEGG IDs and Pathways, respectively\n",
    "        end\n",
    "      end\n",
    "      \n",
    "      if net.interactors.class == Array\n",
    "        net.interactors.each do |int|\n",
    "          unless int.class == String\n",
    "            int.each do |intt|\n",
    "              address = \"http://togows.org/entry/kegg-genes/ath:#{intt}/pathways.json\" #accessing interactor's URLs\n",
    "              response = RestClient::Request.execute(\n",
    "                method: :get,\n",
    "                url: address)\n",
    "              kegg_data = JSON.parse(response.body)\n",
    "              for elem in kegg_data[0].each\n",
    "                if kegg_data[0].empty?\n",
    "                  kegg = []\n",
    "                elsif kegg_data[0]\n",
    "                  kegg << [elem[0],elem[1]] #storing the KEGG IDs and Pathways, respectively\n",
    "                end\n",
    "              end\n",
    "            end\n",
    "          else\n",
    "            unless int == \"direct\"\n",
    "              address = \"http://togows.org/entry/kegg-genes/ath:#{int}/pathways.json\" #accessing interactor's URLs\n",
    "              response = RestClient::Request.execute(\n",
    "                method: :get,\n",
    "                url: address)\n",
    "              kegg_data = JSON.parse(response.body)\n",
    "              for elem in kegg_data[0].each\n",
    "                if kegg_data[0].empty?\n",
    "                  kegg = []\n",
    "                elsif kegg_data[0]\n",
    "                  kegg << [elem[0],elem[1]] #storing the KEGG IDs and Pathways, respectively\n",
    "                end\n",
    "              end\n",
    "            end\n",
    "          end\n",
    "        end\n",
    "      end\n",
    "      kegg = kegg.uniq #deleting duplicated KEGGs\n",
    "      @@allkegg << kegg #storing the KEGG pathways of each network in an array\n",
    "      kegg = []\n",
    "    end\n",
    "  end\n",
    "  \n",
    "  \n",
    "  def Interaction_Network.show_kegg #defining a method to work with KEGG annotations\n",
    "    return @@allkegg\n",
    "  end\n",
    "  \n",
    "  \n",
    "  def Uso_General_Annotation.object_annotation #defining a method to create objects with a network and its GO/KEGG annotations\n",
    "    @@annotated_objects = [] #creating an array to store the Uso_General_Annotation Objects\n",
    "    for i in 0..Interaction_Network.show_all_networks.length-1\n",
    "      @@annotated_objects << Uso_General_Annotation.new(Interaction_Network.show_all_networks[i],Uso_General_Annotation.show_kegg[i],Uso_General_Annotation.show_go[i]) \n",
    "    end\n",
    "  end\n",
    "  \n",
    "  \n",
    "  def Interaction_Network.show_annotated_objects #defining a method to work with Uso_General_Annotation Class objects\n",
    "    return @@annotated_objects\n",
    "  end\n",
    "  \n",
    "  \n",
    "end  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create an âInteractionNetworkâ Object to contain the members of each network and\tannotate it with any KEGG Pathways the interaction network members are part of"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#only run if you haven't run the boxes above\n",
    "require './Interaction_Network.rb' #using Interaction_Network Class\n",
    "require './Uso_General_Annotation.rb' #using Uso_General_Annotation Class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Searching for interactions in BAR database from UToronto...\n",
      "\n",
      "There is no entry for At4g27030 in BAR database from UToronto.\n",
      "There is no entry for At1g21400 in BAR database from UToronto.\n",
      "There is no entry for At5g19120 in BAR database from UToronto.\n",
      "There is no entry for At1g22690 in BAR database from UToronto.\n",
      "There is no entry for At4g09650 in BAR database from UToronto.\n",
      "There is no entry for At5g55620 in BAR database from UToronto.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "#<Errno::ECONNREFUSED: Failed to open TCP connection to bar.utoronto.ca:9090 (Connection refused - connect(2) for \"bar.utoronto.ca\" port 9090)>\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There is no entry for At3g28270 in BAR database from UToronto.\n",
      "There is no entry for At5g64040 in BAR database from UToronto.\n",
      "There is no entry for At1g29660 in BAR database from UToronto.\n",
      "There is no entry for At1g64500 in BAR database from UToronto.\n",
      "There is no entry for At2g20670 in BAR database from UToronto.\n",
      "There is no entry for At3g47470 in BAR database from UToronto.\n",
      "There is no entry for At5g64410 in BAR database from UToronto.\n",
      "There is no entry for At1g15980 in BAR database from UToronto.\n",
      "There is no entry for At1g12250 in BAR database from UToronto.\n",
      "There is no entry for At2g21210 in BAR database from UToronto.\n",
      "There is no entry for At4g28660 in BAR database from UToronto.\n",
      "There is no entry for At1g52230 in BAR database from UToronto.\n",
      "There is no entry for At3g18890 in BAR database from UToronto.\n",
      "There is no entry for At2g04039 in BAR database from UToronto.\n",
      "There is no entry for At3g48420 in BAR database from UToronto.\n",
      "There is no entry for At5g07020 in BAR database from UToronto.\n",
      "There is no entry for At2g34430 in BAR database from UToronto.\n",
      "There is no entry for At1g18060 in BAR database from UToronto.\n",
      "There is no entry for At2g30600 in BAR database from UToronto.\n",
      "There is no entry for At1g65490 in BAR database from UToronto.\n",
      "There is no entry for At3g21670 in BAR database from UToronto.\n",
      "There is no entry for At4g27260 in BAR database from UToronto.\n",
      "There is no entry for At1g55480 in BAR database from UToronto.\n",
      "There is no entry for At1g29395 in BAR database from UToronto.\n",
      "There is no entry for At1g70820 in BAR database from UToronto.\n",
      "There is no entry for At3g26740 in BAR database from UToronto.\n",
      "There is no entry for At1g64680 in BAR database from UToronto.\n",
      "There is no entry for At2g25200 in BAR database from UToronto.\n",
      "There is no entry for At4g21280 in BAR database from UToronto.\n",
      "There is no entry for At1g06680 in BAR database from UToronto.\n",
      "There is no entry for At2g43550 in BAR database from UToronto.\n",
      "There is no entry for At3g02730 in BAR database from UToronto.\n",
      "There is no entry for At2g39470 in BAR database from UToronto.\n",
      "There is no entry for At1g77490 in BAR database from UToronto.\n",
      "There is no entry for At5g01015 in BAR database from UToronto.\n",
      "There is no entry for At1g14150 in BAR database from UToronto.\n",
      "There is no entry for At1g64720 in BAR database from UToronto.\n",
      "There is no entry for At3g56940 in BAR database from UToronto.\n",
      "There is no entry for At4g28030 in BAR database from UToronto.\n",
      "There is no entry for At1g78370 in BAR database from UToronto.\n",
      "There is no entry for At1g23205 in BAR database from UToronto.\n",
      "There is no entry for At1g68010 in BAR database from UToronto.\n",
      "There is no entry for At3g61470 in BAR database from UToronto.\n",
      "There is no entry for At4g28750 in BAR database from UToronto.\n",
      "There is no entry for At5g24210 in BAR database from UToronto.\n",
      "There is no entry for At3g54890 in BAR database from UToronto.\n",
      "There is no entry for At1g72610 in BAR database from UToronto.\n",
      "There is no entry for At3g08940 in BAR database from UToronto.\n",
      "There is no entry for At5g62280 in BAR database from UToronto.\n",
      "There is no entry for At5g18670 in BAR database from UToronto.\n",
      "There is no entry for At1g28330 in BAR database from UToronto.\n",
      "There is no entry for At4g33660 in BAR database from UToronto.\n",
      "There is no entry for At1g76080 in BAR database from UToronto.\n",
      "There is no entry for At3g46780 in BAR database from UToronto.\n",
      "There is no entry for At2g33830 in BAR database from UToronto.\n",
      "There is no entry for At4g02530 in BAR database from UToronto.\n",
      "There is no entry for At1g13930 in BAR database from UToronto.\n",
      "There is no entry for At1g76100 in BAR database from UToronto.\n",
      "There is no entry for At1g08380 in BAR database from UToronto.\n",
      "There is no entry for At2g38210 in BAR database from UToronto.\n",
      "There is no entry for At3g05730 in BAR database from UToronto.\n",
      "There is no entry for At5g48490 in BAR database from UToronto.\n",
      "There is no entry for At1g20440 in BAR database from UToronto.\n",
      "There is no entry for At5g02160 in BAR database from UToronto.\n",
      "There is no entry for At2g05540 in BAR database from UToronto.\n",
      "There is no entry for At3g62550 in BAR database from UToronto.\n",
      "There is no entry for At2g43100 in BAR database from UToronto.\n",
      "There is no entry for At5g49360 in BAR database from UToronto.\n",
      "There is no entry for At1g30380 in BAR database from UToronto.\n",
      "There is no entry for At1g78460 in BAR database from UToronto.\n",
      "There is no entry for At3g16140 in BAR database from UToronto.\n",
      "There is no entry for At4g12310 in BAR database from UToronto.\n",
      "There is no entry for At5g58260 in BAR database from UToronto.\n",
      "\n",
      "Binding networks...\n",
      "\n",
      "\n",
      "Retrieving KEGG pathways...\n",
      "\n",
      "\n",
      "Retrieving GO terms...\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0..150"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Interaction_Network.retrieve_networks #creating objects from 'SubNetwork.txt' (it only contains a few genes, puts $ruby make_report.rb in terminal to see the interactions of the 168 genes) \n",
    "Interaction_Network.binding_networks #connecting the previous objects to make bigger networks and introducing them into new objects\n",
    "Uso_General_Annotation.KEGG_annotation #retrieving KEGG pathways\n",
    "Uso_General_Annotation.GO_annotation #retrieving GO terms\n",
    "Uso_General_Annotation.object_annotation #creating new objects with the final networks and its annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[#<Uso_General_Annotation:0x0000561fbcc0d2a8 @network_members=#<Interaction_Network:0x0000561fbd819b50 @agi_locus1=\"At2g45170\", @agi_locus2=\"At5g54270\", @interactors=[\"At4g37930\", \"At5g26780\"]>, @kegg=[[\"ath04136\", \"Autophagy - other\"], [\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00460\", \"Cyanoamino acid metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00670\", \"One carbon pool by folate\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"], [\"ath01240\", \"Biosynthesis of cofactors\"]], @go=[[\"0006914\", \"autophagy\"], [\"0006995\", \"cellular response to nitrogen starvation\"], [\"0009267\", \"cellular response to starvation\"], [\"0015031\", \"protein transport\"], [\"0045471\", \"response to ethanol\"], [\"0015979\", \"photosynthesis\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0009737\", \"response to abscisic acid\"], [\"0009635\", \"response to herbicide\"], [\"0009644\", \"response to high light intensity\"], [\"0009416\", \"response to light stimulus\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0006412\", \"translation\"], [\"0007623\", \"circadian rhythm\"], [\"0046655\", \"folic acid metabolic process\"], [\"0019264\", \"glycine biosynthetic process from serine\"], [\"0006544\", \"glycine metabolic process\"], [\"0006565\", \"L-serine catabolic process\"], [\"0006563\", \"L-serine metabolic process\"], [\"0006730\", \"one-carbon metabolic process\"], [\"0009853\", \"photorespiration\"], [\"0009626\", \"plant-type hypersensitive response\"], [\"0046686\", \"response to cadmium ion\"], [\"0009409\", \"response to cold\"], [\"0035999\", \"tetrahydrofolate interconversion\"], [\"0046653\", \"tetrahydrofolate metabolic process\"], [\"0032259\", \"methylation\"], [\"0042545\", \"cell wall modification\"], [\"0045490\", \"pectin catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d280 @network_members=#<Interaction_Network:0x0000561fbd819b00 @agi_locus1=\"At5g04140\", @agi_locus2=\"At5g54270\", @interactors=[\"At4g37930\"]>, @kegg=[[\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00460\", \"Cyanoamino acid metabolism\"], [\"ath00670\", \"One carbon pool by folate\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"], [\"ath01240\", \"Biosynthesis of cofactors\"]], @go=[[\"0019676\", \"ammonia assimilation cycle\"], [\"0006537\", \"glutamate biosynthetic process\"], [\"0097054\", \"L-glutamate biosynthetic process\"], [\"0009853\", \"photorespiration\"], [\"0080114\", \"positive regulation of glycine hydroxymethyltransferase activity\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0006541\", \"glutamine metabolic process\"], [\"0015979\", \"photosynthesis\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0009737\", \"response to abscisic acid\"], [\"0009635\", \"response to herbicide\"], [\"0009644\", \"response to high light intensity\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0006412\", \"translation\"], [\"0007623\", \"circadian rhythm\"], [\"0046655\", \"folic acid metabolic process\"], [\"0019264\", \"glycine biosynthetic process from serine\"], [\"0006544\", \"glycine metabolic process\"], [\"0006565\", \"L-serine catabolic process\"], [\"0006563\", \"L-serine metabolic process\"], [\"0006730\", \"one-carbon metabolic process\"], [\"0009626\", \"plant-type hypersensitive response\"], [\"0046686\", \"response to cadmium ion\"], [\"0009409\", \"response to cold\"], [\"0035999\", \"tetrahydrofolate interconversion\"], [\"0046653\", \"tetrahydrofolate metabolic process\"], [\"0032259\", \"methylation\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d258 @network_members=#<Interaction_Network:0x0000561fbd819ab0 @agi_locus1=\"At4g37980\", @agi_locus2=\"At5g54270\", @interactors=[\"At4g34840\"]>, @kegg=[[\"ath00940\", \"Phenylpropanoid biosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath00270\", \"Cysteine and methionine metabolism\"]], @go=[[\"0010597\", \"green leaf volatile biosynthetic process\"], [\"0009809\", \"lignin biosynthetic process\"], [\"0009617\", \"response to bacterium\"], [\"0033512\", \"L-lysine catabolic process to acetyl-CoA via saccharopine\"], [\"0015979\", \"photosynthesis\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0009737\", \"response to abscisic acid\"], [\"0009635\", \"response to herbicide\"], [\"0009644\", \"response to high light intensity\"], [\"0009416\", \"response to light stimulus\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0006412\", \"translation\"], [\"0019509\", \"L-methionine salvage from methylthioadenosine\"], [\"0009116\", \"nucleoside metabolic process\"], [\"0010087\", \"phloem or xylem histogenesis\"], [\"0000003\", \"reproduction\"], [\"0009627\", \"systemic acquired resistance\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d230 @network_members=#<Interaction_Network:0x0000561fbd819a88 @agi_locus1=\"At2g34420\", @agi_locus2=\"At5g54270\", @interactors=[\"At4g37140\", \"At1g79870\", \"At3g44680\", \"At4g34840\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath00270\", \"Cysteine and methionine metabolism\"]], @go=[[\"0009765\", \"photosynthesis, light harvesting\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009416\", \"response to light stimulus\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0015979\", \"photosynthesis\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0009737\", \"response to abscisic acid\"], [\"0009635\", \"response to herbicide\"], [\"0009644\", \"response to high light intensity\"], [\"0006412\", \"translation\"], [\"0009793\", \"embryo development ending in seed dormancy\"], [\"0009854\", \"oxidative photosynthetic carbon pathway\"], [\"1990619\", \"histone H3-K9 deacetylation\"], [\"0010187\", \"negative regulation of seed germination\"], [\"1900055\", \"regulation of leaf senescence\"], [\"0050832\", \"defense response to fungus\"], [\"0009626\", \"plant-type hypersensitive response\"], [\"0019509\", \"L-methionine salvage from methylthioadenosine\"], [\"0009116\", \"nucleoside metabolic process\"], [\"0010087\", \"phloem or xylem histogenesis\"], [\"0000003\", \"reproduction\"], [\"0009627\", \"systemic acquired resistance\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d208 @network_members=#<Interaction_Network:0x0000561fbd819a60 @agi_locus1=\"At3g27690\", @agi_locus2=\"At5g54270\", @interactors=[\"At5g12980\", \"At5g45870\", \"At1g79870\", \"At4g34840\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath03018\", \"RNA degradation\"], [\"ath04016\", \"MAPK signaling pathway - plant\"], [\"ath04075\", \"Plant hormone signal transduction\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath00270\", \"Cysteine and methionine metabolism\"]], @go=[[\"0045893\", \"positive regulation of transcription, DNA-templated\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0071215\", \"cellular response to abscisic acid stimulus\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"1903428\", \"positive regulation of reactive oxygen species biosynthetic process\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0090333\", \"regulation of stomatal closure\"], [\"0009409\", \"response to cold\"], [\"0009269\", \"response to desiccation\"], [\"0010218\", \"response to far red light\"], [\"0009644\", \"response to high light intensity\"], [\"0009416\", \"response to light stimulus\"], [\"0009645\", \"response to low light intensity stimulus\"], [\"0010114\", \"response to red light\"], [\"0030104\", \"water homeostasis\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0015979\", \"photosynthesis\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0009737\", \"response to abscisic acid\"], [\"0009635\", \"response to herbicide\"], [\"0006412\", \"translation\"], [\"0017148\", \"negative regulation of translation\"], [\"0009738\", \"abscisic acid-activated signaling pathway\"], [\"0080163\", \"regulation of protein serine/threonine phosphatase activity\"], [\"0009854\", \"oxidative photosynthetic carbon pathway\"], [\"0019509\", \"L-methionine salvage from methylthioadenosine\"], [\"0009116\", \"nucleoside metabolic process\"], [\"0010087\", \"phloem or xylem histogenesis\"], [\"0000003\", \"reproduction\"], [\"0009627\", \"systemic acquired resistance\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d1e0 @network_members=#<Interaction_Network:0x0000561fbd819a38 @agi_locus1=\"At5g35630\", @agi_locus2=\"At5g54270\", @interactors=[\"At1g79870\"]>, @kegg=[[\"ath00220\", \"Arginine biosynthesis\"], [\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01230\", \"Biosynthesis of amino acids\"], [\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0007568\", \"aging\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006542\", \"glutamine biosynthetic process\"], [\"0046686\", \"response to cadmium ion\"], [\"0006807\", \"nitrogen compound metabolic process\"], [\"0015979\", \"photosynthesis\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0009737\", \"response to abscisic acid\"], [\"0009635\", \"response to herbicide\"], [\"0009644\", \"response to high light intensity\"], [\"0009416\", \"response to light stimulus\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0006412\", \"translation\"], [\"0009854\", \"oxidative photosynthetic carbon pathway\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d1b8 @network_members=#<Interaction_Network:0x0000561fbd8199e8 @agi_locus1=\"At2g45170\", @agi_locus2=\"At5g04140\", @interactors=[\"At4g37930\"]>, @kegg=[[\"ath04136\", \"Autophagy - other\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00460\", \"Cyanoamino acid metabolism\"], [\"ath00670\", \"One carbon pool by folate\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"], [\"ath01240\", \"Biosynthesis of cofactors\"]], @go=[[\"0006914\", \"autophagy\"], [\"0006995\", \"cellular response to nitrogen starvation\"], [\"0009267\", \"cellular response to starvation\"], [\"0015031\", \"protein transport\"], [\"0045471\", \"response to ethanol\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006537\", \"glutamate biosynthetic process\"], [\"0097054\", \"L-glutamate biosynthetic process\"], [\"0009853\", \"photorespiration\"], [\"0080114\", \"positive regulation of glycine hydroxymethyltransferase activity\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0006541\", \"glutamine metabolic process\"], [\"0007623\", \"circadian rhythm\"], [\"0046655\", \"folic acid metabolic process\"], [\"0019264\", \"glycine biosynthetic process from serine\"], [\"0006544\", \"glycine metabolic process\"], [\"0006565\", \"L-serine catabolic process\"], [\"0006563\", \"L-serine metabolic process\"], [\"0006730\", \"one-carbon metabolic process\"], [\"0009626\", \"plant-type hypersensitive response\"], [\"0046686\", \"response to cadmium ion\"], [\"0009409\", \"response to cold\"], [\"0035999\", \"tetrahydrofolate interconversion\"], [\"0046653\", \"tetrahydrofolate metabolic process\"], [\"0032259\", \"methylation\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d190 @network_members=#<Interaction_Network:0x0000561fbd819998 @agi_locus1=\"At1g23740\", @agi_locus2=\"At2g45170\", @interactors=[\"At4g33710\", \"At1g07180\"]>, @kegg=[[\"ath04136\", \"Autophagy - other\"], [\"ath04016\", \"MAPK signaling pathway - plant\"], [\"ath04075\", \"Plant hormone signal transduction\"], [\"ath04626\", \"Plant-pathogen interaction\"]], @go=[[\"0009409\", \"response to cold\"], [\"0046856\", \"phosphatidylinositol dephosphorylation\"], [\"0006914\", \"autophagy\"], [\"0006995\", \"cellular response to nitrogen starvation\"], [\"0009267\", \"cellular response to starvation\"], [\"0015031\", \"protein transport\"], [\"0045471\", \"response to ethanol\"], [\"0006952\", \"defense response\"], [\"0071482\", \"cellular response to light stimulus\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d168 @network_members=#<Interaction_Network:0x0000561fbd819948 @agi_locus1=\"At2g45170\", @agi_locus2=\"At4g21210\", @interactors=[\"At2g33210\"]>, @kegg=[[\"ath04136\", \"Autophagy - other\"], [\"ath03018\", \"RNA degradation\"]], @go=[[\"0006914\", \"autophagy\"], [\"0006995\", \"cellular response to nitrogen starvation\"], [\"0009267\", \"cellular response to starvation\"], [\"0015031\", \"protein transport\"], [\"0045471\", \"response to ethanol\"], [\"0006457\", \"protein folding\"], [\"0042026\", \"protein refolding\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d140 @network_members=#<Interaction_Network:0x0000561fbd819920 @agi_locus1=\"At1g12900\", @agi_locus2=\"At2g45170\", @interactors=[\"At1g25155\"]>, @kegg=[[\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath04136\", \"Autophagy - other\"], [\"ath00400\", \"Phenylalanine, tyrosine and tryptophan biosynthesis\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01230\", \"Biosynthesis of amino acids\"]], @go=[[\"0006006\", \"glucose metabolic process\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0008643\", \"carbohydrate transport\"], [\"0006914\", \"autophagy\"], [\"0006995\", \"cellular response to nitrogen starvation\"], [\"0009267\", \"cellular response to starvation\"], [\"0015031\", \"protein transport\"], [\"0045471\", \"response to ethanol\"], [\"0006541\", \"glutamine metabolic process\"], [\"0000162\", \"tryptophan biosynthetic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d118 @network_members=#<Interaction_Network:0x0000561fbd8198f8 @agi_locus1=\"At2g26740\", @agi_locus2=\"At2g45170\", @interactors=[\"At5g14060\"]>, @kegg=[[\"ath04136\", \"Autophagy - other\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00261\", \"Monobactam biosynthesis\"], [\"ath00270\", \"Cysteine and methionine metabolism\"], [\"ath00300\", \"Lysine biosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01210\", \"2-Oxocarboxylic acid metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"]], @go=[[\"0009733\", \"response to auxin\"], [\"0009414\", \"response to water deprivation\"], [\"0006914\", \"autophagy\"], [\"0006995\", \"cellular response to nitrogen starvation\"], [\"0009267\", \"cellular response to starvation\"], [\"0015031\", \"protein transport\"], [\"0045471\", \"response to ethanol\"], [\"0009090\", \"homoserine biosynthetic process\"], [\"0009089\", \"lysine biosynthetic process via diaminopimelate\"], [\"0009088\", \"threonine biosynthetic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d0f0 @network_members=#<Interaction_Network:0x0000561fbd8198d0 @agi_locus1=\"At1g80440\", @agi_locus2=\"At4g39710\", @interactors=[\"At4g13530\"]>, @kegg=[], @go=[[\"0080037\", \"negative regulation of cytokinin-activated signaling pathway\"], [\"2000762\", \"regulation of phenylpropanoid metabolic process\"], [\"0006952\", \"defense response\"], [\"0006355\", \"regulation of transcription, DNA-templated\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d0c8 @network_members=#<Interaction_Network:0x0000561fbd819880 @agi_locus1=\"At4g17090\", @agi_locus2=\"At5g04140\", @interactors=[\"At4g34230\", \"At2g21730\"]>, @kegg=[[\"ath00500\", \"Starch and sucrose metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath00940\", \"Phenylpropanoid biosynthesis\"]], @go=[[\"0000024\", \"maltose biosynthetic process\"], [\"0009409\", \"response to cold\"], [\"0005983\", \"starch catabolic process\"], [\"0000272\", \"polysaccharide catabolic process\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006537\", \"glutamate biosynthetic process\"], [\"0097054\", \"L-glutamate biosynthetic process\"], [\"0009853\", \"photorespiration\"], [\"0080114\", \"positive regulation of glycine hydroxymethyltransferase activity\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0006541\", \"glutamine metabolic process\"], [\"0009809\", \"lignin biosynthetic process\"], [\"0009088\", \"threonine biosynthetic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d0a0 @network_members=#<Interaction_Network:0x0000561fbd819830 @agi_locus1=\"At4g37980\", @agi_locus2=\"At5g04140\", @interactors=\"direct\">, @kegg=[[\"ath00940\", \"Phenylpropanoid biosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"]], @go=[[\"0010597\", \"green leaf volatile biosynthetic process\"], [\"0009809\", \"lignin biosynthetic process\"], [\"0009617\", \"response to bacterium\"], [\"0033512\", \"L-lysine catabolic process to acetyl-CoA via saccharopine\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006537\", \"glutamate biosynthetic process\"], [\"0097054\", \"L-glutamate biosynthetic process\"], [\"0009853\", \"photorespiration\"], [\"0080114\", \"positive regulation of glycine hydroxymethyltransferase activity\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0006541\", \"glutamine metabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d078 @network_members=#<Interaction_Network:0x0000561fbd8197e0 @agi_locus1=\"At3g27690\", @agi_locus2=\"At5g04140\", @interactors=[\"At1g01480\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath00270\", \"Cysteine and methionine metabolism\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath04016\", \"MAPK signaling pathway - plant\"]], @go=[[\"0045893\", \"positive regulation of transcription, DNA-templated\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0071215\", \"cellular response to abscisic acid stimulus\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"1903428\", \"positive regulation of reactive oxygen species biosynthetic process\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0090333\", \"regulation of stomatal closure\"], [\"0009409\", \"response to cold\"], [\"0009269\", \"response to desiccation\"], [\"0010218\", \"response to far red light\"], [\"0009644\", \"response to high light intensity\"], [\"0009416\", \"response to light stimulus\"], [\"0009645\", \"response to low light intensity stimulus\"], [\"0010114\", \"response to red light\"], [\"0030104\", \"water homeostasis\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006537\", \"glutamate biosynthetic process\"], [\"0097054\", \"L-glutamate biosynthetic process\"], [\"0009853\", \"photorespiration\"], [\"0080114\", \"positive regulation of glycine hydroxymethyltransferase activity\"], [\"0009744\", \"response to sucrose\"], [\"0006541\", \"glutamine metabolic process\"], [\"0009693\", \"ethylene biosynthetic process\"], [\"0009835\", \"fruit ripening\"], [\"0009058\", \"biosynthetic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d050 @network_members=#<Interaction_Network:0x0000561fbd8197b8 @agi_locus1=\"At2g26740\", @agi_locus2=\"At5g04140\", @interactors=[\"At4g39330\"]>, @kegg=[[\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath00940\", \"Phenylpropanoid biosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"]], @go=[[\"0009733\", \"response to auxin\"], [\"0009414\", \"response to water deprivation\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006537\", \"glutamate biosynthetic process\"], [\"0097054\", \"L-glutamate biosynthetic process\"], [\"0009853\", \"photorespiration\"], [\"0080114\", \"positive regulation of glycine hydroxymethyltransferase activity\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0006541\", \"glutamine metabolic process\"], [\"0009809\", \"lignin biosynthetic process\"], [\"0007186\", \"G protein-coupled receptor signaling pathway\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d028 @network_members=#<Interaction_Network:0x0000561fbd819790 @agi_locus1=\"At1g15820\", @agi_locus2=\"At2g34420\", @interactors=[\"At3g26090\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0010196\", \"nonphotochemical quenching\"], [\"0015979\", \"photosynthesis\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0009867\", \"jasmonic acid mediated signaling pathway\"], [\"0009968\", \"negative regulation of signal transduction\"], [\"0043547\", \"positive regulation of GTPase activity\"], [\"0042127\", \"regulation of cell population proliferation\"], [\"0009737\", \"response to abscisic acid\"], [\"0009749\", \"response to glucose\"], [\"0009414\", \"response to water deprivation\"], [\"0010182\", \"sugar mediated signaling pathway\"], [\"0006355\", \"regulation of transcription, DNA-templated\"]]>, #<Uso_General_Annotation:0x0000561fbcc0d000 @network_members=#<Interaction_Network:0x0000561fbd819740 @agi_locus1=\"At1g15820\", @agi_locus2=\"At1g29920\", @interactors=[\"At3g26090\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0010196\", \"nonphotochemical quenching\"], [\"0015979\", \"photosynthesis\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0009750\", \"response to fructose\"], [\"0009867\", \"jasmonic acid mediated signaling pathway\"], [\"0009968\", \"negative regulation of signal transduction\"], [\"0043547\", \"positive regulation of GTPase activity\"], [\"0042127\", \"regulation of cell population proliferation\"], [\"0009737\", \"response to abscisic acid\"], [\"0009749\", \"response to glucose\"], [\"0009414\", \"response to water deprivation\"], [\"0010182\", \"sugar mediated signaling pathway\"], [\"0006355\", \"regulation of transcription, DNA-templated\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cfd8 @network_members=#<Interaction_Network:0x0000561fbd819718 @agi_locus1=\"At1g15820\", @agi_locus2=\"At5g01530\", @interactors=[\"At3g26090\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0010196\", \"nonphotochemical quenching\"], [\"0015979\", \"photosynthesis\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0009867\", \"jasmonic acid mediated signaling pathway\"], [\"0009968\", \"negative regulation of signal transduction\"], [\"0043547\", \"positive regulation of GTPase activity\"], [\"0042127\", \"regulation of cell population proliferation\"], [\"0009737\", \"response to abscisic acid\"], [\"0009749\", \"response to glucose\"], [\"0009414\", \"response to water deprivation\"], [\"0010182\", \"sugar mediated signaling pathway\"], [\"0006355\", \"regulation of transcription, DNA-templated\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cfb0 @network_members=#<Interaction_Network:0x0000561fbd8196f0 @agi_locus1=\"At1g03130\", @agi_locus2=\"At2g17880\", @interactors=[\"At3g48150\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"]], @go=[[\"0015979\", \"photosynthesis\"], [\"0031145\", \"anaphase-promoting complex-dependent catabolic process\"], [\"0051301\", \"cell division\"], [\"0007091\", \"metaphase/anaphase transition of mitotic cell cycle\"], [\"0045842\", \"positive regulation of mitotic metaphase/anaphase transition\"], [\"0016567\", \"protein ubiquitination\"], [\"0031347\", \"regulation of defense response\"], [\"0007049\", \"cell cycle\"], [\"0030071\", \"regulation of mitotic metaphase/anaphase transition\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cf88 @network_members=#<Interaction_Network:0x0000561fbd8196c8 @agi_locus1=\"At1g03130\", @agi_locus2=\"At2g34420\", @interactors=[\"At3g47620\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00196\", \"Photosynthesis - antenna proteins\"]], @go=[[\"0015979\", \"photosynthesis\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009416\", \"response to light stimulus\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0010229\", \"inflorescence development\"], [\"0031347\", \"regulation of defense response\"], [\"0010029\", \"regulation of seed germination\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cf60 @network_members=#<Interaction_Network:0x0000561fbd819678 @agi_locus1=\"At1g03130\", @agi_locus2=\"At3g16250\", @interactors=[\"At3g48150\", \"At3g02150\", \"At3g47620\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"]], @go=[[\"0015979\", \"photosynthesis\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0031145\", \"anaphase-promoting complex-dependent catabolic process\"], [\"0051301\", \"cell division\"], [\"0007091\", \"metaphase/anaphase transition of mitotic cell cycle\"], [\"0045842\", \"positive regulation of mitotic metaphase/anaphase transition\"], [\"0016567\", \"protein ubiquitination\"], [\"0031347\", \"regulation of defense response\"], [\"0007049\", \"cell cycle\"], [\"0030071\", \"regulation of mitotic metaphase/anaphase transition\"], [\"0048366\", \"leaf development\"], [\"0010150\", \"leaf senescence\"], [\"2000032\", \"regulation of secondary shoot formation\"], [\"0006021\", \"inositol biosynthetic process\"], [\"0046855\", \"inositol phosphate dephosphorylation\"], [\"0046854\", \"phosphatidylinositol phosphorylation\"], [\"0010229\", \"inflorescence development\"], [\"0010029\", \"regulation of seed germination\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cf38 @network_members=#<Interaction_Network:0x0000561fbd819628 @agi_locus1=\"At1g03130\", @agi_locus2=\"At1g29910\", @interactors=[\"At3g47620\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00196\", \"Photosynthesis - antenna proteins\"]], @go=[[\"0015979\", \"photosynthesis\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009750\", \"response to fructose\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0009733\", \"response to auxin\"], [\"0010229\", \"inflorescence development\"], [\"0031347\", \"regulation of defense response\"], [\"0010029\", \"regulation of seed germination\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cf10 @network_members=#<Interaction_Network:0x0000561fbd8195d8 @agi_locus1=\"At1g03130\", @agi_locus2=\"At1g29920\", @interactors=[\"At3g47620\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00196\", \"Photosynthesis - antenna proteins\"]], @go=[[\"0015979\", \"photosynthesis\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009750\", \"response to fructose\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0010229\", \"inflorescence development\"], [\"0031347\", \"regulation of defense response\"], [\"0010029\", \"regulation of seed germination\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cee8 @network_members=#<Interaction_Network:0x0000561fbd8195b0 @agi_locus1=\"At1g03130\", @agi_locus2=\"At4g02770\", @interactors=[\"At3g48150\", \"At4g29930\", \"At5g61010\", \"At4g19700\", \"At5g32440\", \"At2g23290\", \"At3g47620\", \"At5g37670\", \"At3g08530\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"], [\"ath04144\", \"Endocytosis\"]], @go=[[\"0015979\", \"photosynthesis\"], [\"0140547\", \"acquisition of seed longevity\"], [\"0035556\", \"intracellular signal transduction\"], [\"0031145\", \"anaphase-promoting complex-dependent catabolic process\"], [\"0051301\", \"cell division\"], [\"0007091\", \"metaphase/anaphase transition of mitotic cell cycle\"], [\"0045842\", \"positive regulation of mitotic metaphase/anaphase transition\"], [\"0016567\", \"protein ubiquitination\"], [\"0031347\", \"regulation of defense response\"], [\"0007049\", \"cell cycle\"], [\"0030071\", \"regulation of mitotic metaphase/anaphase transition\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0048268\", \"clathrin coat assembly\"], [\"0006897\", \"endocytosis\"], [\"0052542\", \"defense response by callose deposition\"], [\"0006887\", \"exocytosis\"], [\"0006955\", \"immune response\"], [\"1903553\", \"positive regulation of extracellular exosome assembly\"], [\"0015031\", \"protein transport\"], [\"1903533\", \"regulation of protein targeting\"], [\"0042742\", \"defense response to bacterium\"], [\"0050832\", \"defense response to fungus\"], [\"2000117\", \"negative regulation of cysteine-type endopeptidase activity\"], [\"0043069\", \"negative regulation of programmed cell death\"], [\"0043161\", \"proteasome-mediated ubiquitin-dependent protein catabolic process\"], [\"0043067\", \"regulation of programmed cell death\"], [\"0009739\", \"response to gibberellin\"], [\"0010229\", \"inflorescence development\"], [\"0010029\", \"regulation of seed germination\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0071456\", \"cellular response to hypoxia\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0042542\", \"response to hydrogen peroxide\"], [\"0000302\", \"response to reactive oxygen species\"], [\"0009651\", \"response to salt stress\"], [\"0006886\", \"intracellular protein transport\"], [\"0006898\", \"receptor-mediated endocytosis\"], [\"0016192\", \"vesicle-mediated transport\"], [\"0006606\", \"protein import into nucleus\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cec0 @network_members=#<Interaction_Network:0x0000561fbd819588 @agi_locus1=\"At4g17090\", @agi_locus2=\"At5g65010\", @interactors=[\"At5g39320\"]>, @kegg=[[\"ath00500\", \"Starch and sucrose metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"], [\"ath00040\", \"Pentose and glucuronate interconversions\"], [\"ath00053\", \"Ascorbate and aldarate metabolism\"], [\"ath00520\", \"Amino sugar and nucleotide sugar metabolism\"], [\"ath01240\", \"Biosynthesis of cofactors\"], [\"ath01250\", \"Biosynthesis of nucleotide sugars\"]], @go=[[\"0000024\", \"maltose biosynthetic process\"], [\"0009409\", \"response to cold\"], [\"0005983\", \"starch catabolic process\"], [\"0000272\", \"polysaccharide catabolic process\"], [\"0097164\", \"ammonium ion metabolic process\"], [\"0006529\", \"asparagine biosynthetic process\"], [\"0006541\", \"glutamine metabolic process\"], [\"0042538\", \"hyperosmotic salinity response\"], [\"0070981\", \"L-asparagine biosynthetic process\"], [\"0006024\", \"glycosaminoglycan biosynthetic process\"], [\"0006065\", \"UDP-glucuronate biosynthetic process\"], [\"0032434\", \"regulation of proteasomal ubiquitin-dependent protein catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0ce98 @network_members=#<Interaction_Network:0x0000561fbd819560 @agi_locus1=\"At4g17090\", @agi_locus2=\"At4g37800\", @interactors=[\"At3g26810\"]>, @kegg=[[\"ath00500\", \"Starch and sucrose metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"]], @go=[[\"0000024\", \"maltose biosynthetic process\"], [\"0009409\", \"response to cold\"], [\"0005983\", \"starch catabolic process\"], [\"0000272\", \"polysaccharide catabolic process\"], [\"0042546\", \"cell wall biogenesis\"], [\"0071555\", \"cell wall organization\"], [\"0010411\", \"xyloglucan metabolic process\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009734\", \"auxin-activated signaling pathway\"], [\"0006952\", \"defense response\"], [\"0010152\", \"pollen maturation\"], [\"0016567\", \"protein ubiquitination\"], [\"0031146\", \"SCF-dependent proteasomal ubiquitin-dependent protein catabolic process\"], [\"0048443\", \"stamen development\"], [\"0010089\", \"xylem development\"]]>, #<Uso_General_Annotation:0x0000561fbcc0ce70 @network_members=#<Interaction_Network:0x0000561fbd819510 @agi_locus1=\"At1g12900\", @agi_locus2=\"At4g17090\", @interactors=[\"At5g11920\"]>, @kegg=[[\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath00500\", \"Starch and sucrose metabolism\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath00052\", \"Galactose metabolism\"]], @go=[[\"0006006\", \"glucose metabolic process\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0008643\", \"carbohydrate transport\"], [\"0000024\", \"maltose biosynthetic process\"], [\"0009409\", \"response to cold\"], [\"0005983\", \"starch catabolic process\"], [\"0000272\", \"polysaccharide catabolic process\"], [\"0005975\", \"carbohydrate metabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0ce48 @network_members=#<Interaction_Network:0x0000561fbd8194e8 @agi_locus1=\"At2g26740\", @agi_locus2=\"At4g17090\", @interactors=[\"At5g63860\", \"At2g24270\", \"At1g12820\"]>, @kegg=[[\"ath00500\", \"Starch and sucrose metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath00010\", \"Glycolysis / Gluconeogenesis\"], [\"ath00030\", \"Pentose phosphate pathway\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0009733\", \"response to auxin\"], [\"0009414\", \"response to water deprivation\"], [\"0000024\", \"maltose biosynthetic process\"], [\"0009409\", \"response to cold\"], [\"0005983\", \"starch catabolic process\"], [\"0000272\", \"polysaccharide catabolic process\"], [\"0009649\", \"entrainment of circadian clock\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009411\", \"response to UV\"], [\"0010224\", \"response to UV-B\"], [\"0009734\", \"auxin-activated signaling pathway\"], [\"0071249\", \"cellular response to nitrate\"], [\"0006952\", \"defense response\"], [\"0048527\", \"lateral root development\"], [\"0010152\", \"pollen maturation\"], [\"0080022\", \"primary root development\"], [\"0016567\", \"protein ubiquitination\"], [\"0031146\", \"SCF-dependent proteasomal ubiquitin-dependent protein catabolic process\"], [\"0048443\", \"stamen development\"]]>, #<Uso_General_Annotation:0x0000561fbcc0ce20 @network_members=#<Interaction_Network:0x0000561fbd8194c0 @agi_locus1=\"At3g47340\", @agi_locus2=\"At4g17090\", @interactors=[\"At2g32290\"]>, @kegg=[[\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01230\", \"Biosynthesis of amino acids\"], [\"ath00500\", \"Starch and sucrose metabolism\"]], @go=[[\"0006529\", \"asparagine biosynthetic process\"], [\"0009063\", \"cellular amino acid catabolic process\"], [\"0043617\", \"cellular response to sucrose starvation\"], [\"0006541\", \"glutamine metabolic process\"], [\"0070981\", \"L-asparagine biosynthetic process\"], [\"0009646\", \"response to absence of light\"], [\"0009750\", \"response to fructose\"], [\"0009749\", \"response to glucose\"], [\"0009744\", \"response to sucrose\"], [\"0006096\", \"glycolytic process\"], [\"0000024\", \"maltose biosynthetic process\"], [\"0009409\", \"response to cold\"], [\"0005983\", \"starch catabolic process\"], [\"0000272\", \"polysaccharide catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cdf8 @network_members=#<Interaction_Network:0x0000561fbd819470 @agi_locus1=\"At4g17090\", @agi_locus2=\"At5g48300\", @interactors=[\"At3g47010\"]>, @kegg=[[\"ath00500\", \"Starch and sucrose metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath00520\", \"Amino sugar and nucleotide sugar metabolism\"], [\"ath01250\", \"Biosynthesis of nucleotide sugars\"], [\"ath00460\", \"Cyanoamino acid metabolism\"], [\"ath00940\", \"Phenylpropanoid biosynthesis\"]], @go=[[\"0000024\", \"maltose biosynthetic process\"], [\"0009409\", \"response to cold\"], [\"0005983\", \"starch catabolic process\"], [\"0000272\", \"polysaccharide catabolic process\"], [\"0005978\", \"glycogen biosynthetic process\"], [\"0048573\", \"photoperiodism, flowering\"], [\"0019252\", \"starch biosynthetic process\"], [\"0006457\", \"protein folding\"], [\"0005975\", \"carbohydrate metabolic process\"], [\"0009251\", \"glucan catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cdd0 @network_members=#<Interaction_Network:0x0000561fbd8193f8 @agi_locus1=\"At1g23740\", @agi_locus2=\"At2g45170\", @interactors=[\"At1g07180\", \"At4g33710\"]>, @kegg=[[\"ath04136\", \"Autophagy - other\"], [\"ath04016\", \"MAPK signaling pathway - plant\"], [\"ath04075\", \"Plant hormone signal transduction\"], [\"ath04626\", \"Plant-pathogen interaction\"]], @go=[[\"0009409\", \"response to cold\"], [\"0046856\", \"phosphatidylinositol dephosphorylation\"], [\"0006914\", \"autophagy\"], [\"0006995\", \"cellular response to nitrogen starvation\"], [\"0009267\", \"cellular response to starvation\"], [\"0015031\", \"protein transport\"], [\"0045471\", \"response to ethanol\"], [\"0071482\", \"cellular response to light stimulus\"], [\"0006952\", \"defense response\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cda8 @network_members=#<Interaction_Network:0x0000561fbd8193a8 @agi_locus1=\"At1g23740\", @agi_locus2=\"At4g12830\", @interactors=[\"At5g52640\"]>, @kegg=[[\"ath04141\", \"Protein processing in endoplasmic reticulum\"], [\"ath04626\", \"Plant-pathogen interaction\"]], @go=[[\"0009409\", \"response to cold\"], [\"0046856\", \"phosphatidylinositol dephosphorylation\"], [\"0034605\", \"cellular response to heat\"], [\"0061077\", \"chaperone-mediated protein folding\"], [\"0042742\", \"defense response to bacterium\"], [\"0045087\", \"innate immune response\"], [\"0006457\", \"protein folding\"], [\"0050821\", \"protein stabilization\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cd80 @network_members=#<Interaction_Network:0x0000561fbd819380 @agi_locus1=\"At1g23740\", @agi_locus2=\"At2g26080\", @interactors=[\"At3g06470\", \"At1g02730\"]>, @kegg=[[\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath00520\", \"Amino sugar and nucleotide sugar metabolism\"]], @go=[[\"0009409\", \"response to cold\"], [\"0046856\", \"phosphatidylinositol dephosphorylation\"], [\"0019464\", \"glycine decarboxylation via glycine cleavage system\"], [\"0006546\", \"glycine catabolic process\"], [\"0000398\", \"mRNA splicing, via spliceosome\"], [\"0042761\", \"very long-chain fatty acid biosynthetic process\"], [\"0006629\", \"lipid metabolic process\"], [\"0034625\", \"fatty acid elongation, monounsaturated fatty acid\"], [\"0034626\", \"fatty acid elongation, polyunsaturated fatty acid\"], [\"0019367\", \"fatty acid elongation, saturated fatty acid\"], [\"0030148\", \"sphingolipid biosynthetic process\"], [\"0006633\", \"fatty acid biosynthetic process\"], [\"0000919\", \"cell plate assembly\"], [\"0042546\", \"cell wall biogenesis\"], [\"0071555\", \"cell wall organization\"], [\"0030244\", \"cellulose biosynthetic process\"], [\"0009833\", \"plant-type primary cell wall biogenesis\"], [\"0006970\", \"response to osmotic stress\"], [\"0009651\", \"response to salt stress\"], [\"0009414\", \"response to water deprivation\"], [\"0048367\", \"shoot system development\"], [\"0006782\", \"protoporphyrinogen IX biosynthetic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cd58 @network_members=#<Interaction_Network:0x0000561fbd819358 @agi_locus1=\"At1g23740\", @agi_locus2=\"At3g45780\", @interactors=[\"At3g08910\"]>, @kegg=[], @go=[[\"0009409\", \"response to cold\"], [\"0046856\", \"phosphatidylinositol dephosphorylation\"], [\"0009904\", \"chloroplast accumulation movement\"], [\"0009903\", \"chloroplast avoidance movement\"], [\"0007623\", \"circadian rhythm\"], [\"0010362\", \"negative regulation of anion channel activity by blue light\"], [\"0009638\", \"phototropism\"], [\"0046777\", \"protein autophosphorylation\"], [\"0006468\", \"protein phosphorylation\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0010155\", \"regulation of proton transport\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0009637\", \"response to blue light\"], [\"0050896\", \"response to stimulus\"], [\"0051085\", \"chaperone cofactor-dependent protein refolding\"], [\"0006457\", \"protein folding\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cd30 @network_members=#<Interaction_Network:0x0000561fbd819308 @agi_locus1=\"At2g46340\", @agi_locus2=\"At5g15850\", @interactors=[[\"At1g53090\"], \"direct\"]>, @kegg=[[\"ath04712\", \"Circadian rhythm - plant\"]], @go=[[\"0009658\", \"chloroplast organization\"], [\"0010100\", \"negative regulation of photomorphogenesis\"], [\"0009640\", \"photomorphogenesis\"], [\"0010017\", \"red or far-red light signaling pathway\"], [\"0009585\", \"red, far-red light phototransduction\"], [\"2000028\", \"regulation of photoperiodism, flowering\"], [\"0009637\", \"response to blue light\"], [\"0010218\", \"response to far red light\"], [\"0010114\", \"response to red light\"], [\"0048575\", \"short-day photoperiodism, flowering\"], [\"0009909\", \"regulation of flower development\"], [\"0000398\", \"mRNA splicing, via spliceosome\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cd08 @network_members=#<Interaction_Network:0x0000561fbd8192e0 @agi_locus1=\"At2g21330\", @agi_locus2=\"At4g37800\", @interactors=[\"At4g32520\"]>, @kegg=[[\"ath00010\", \"Glycolysis / Gluconeogenesis\"], [\"ath00030\", \"Pentose phosphate pathway\"], [\"ath00051\", \"Fructose and mannose metabolism\"], [\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00460\", \"Cyanoamino acid metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00670\", \"One carbon pool by folate\"], [\"ath01240\", \"Biosynthesis of cofactors\"]], @go=[[\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0006096\", \"glycolytic process\"], [\"0046686\", \"response to cadmium ion\"], [\"0071555\", \"cell wall organization\"], [\"0030244\", \"cellulose biosynthetic process\"], [\"0042546\", \"cell wall biogenesis\"], [\"0010411\", \"xyloglucan metabolic process\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0046655\", \"folic acid metabolic process\"], [\"0019264\", \"glycine biosynthetic process from serine\"], [\"0006565\", \"L-serine catabolic process\"], [\"0006730\", \"one-carbon metabolic process\"], [\"0035999\", \"tetrahydrofolate interconversion\"], [\"0046653\", \"tetrahydrofolate metabolic process\"], [\"0019953\", \"sexual reproduction\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cce0 @network_members=#<Interaction_Network:0x0000561fbd8192b8 @agi_locus1=\"At2g21330\", @agi_locus2=\"At2g26740\", @interactors=[\"At4g20360\"]>, @kegg=[[\"ath00010\", \"Glycolysis / Gluconeogenesis\"], [\"ath00030\", \"Pentose phosphate pathway\"], [\"ath00051\", \"Fructose and mannose metabolism\"], [\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"], [\"ath04626\", \"Plant-pathogen interaction\"]], @go=[[\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0006096\", \"glycolytic process\"], [\"0046686\", \"response to cadmium ion\"], [\"0071555\", \"cell wall organization\"], [\"0030244\", \"cellulose biosynthetic process\"], [\"0009733\", \"response to auxin\"], [\"0009414\", \"response to water deprivation\"], [\"0009658\", \"chloroplast organization\"], [\"0048366\", \"leaf development\"], [\"0070125\", \"mitochondrial translational elongation\"], [\"0006414\", \"translational elongation\"], [\"0006421\", \"asparaginyl-tRNA aminoacylation\"]]>, #<Uso_General_Annotation:0x0000561fbcc0ccb8 @network_members=#<Interaction_Network:0x0000561fbd819290 @agi_locus1=\"At3g01500\", @agi_locus2=\"At4g22890\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0015976\", \"carbon utilization\"], [\"0042742\", \"defense response to bacterium\"], [\"0050832\", \"defense response to fungus\"], [\"2000122\", \"negative regulation of stomatal complex development\"], [\"0015979\", \"photosynthesis\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0010037\", \"response to carbon dioxide\"], [\"0009409\", \"response to cold\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cc90 @network_members=#<Interaction_Network:0x0000561fbd819240 @agi_locus1=\"At4g22890\", @agi_locus2=\"At5g14740\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0015976\", \"carbon utilization\"], [\"0042742\", \"defense response to bacterium\"], [\"0006865\", \"amino acid transport\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cc68 @network_members=#<Interaction_Network:0x0000561fbd8191f0 @agi_locus1=\"At2g39730\", @agi_locus2=\"At4g22890\", @interactors=[\"At5g42980\"]>, @kegg=[], @go=[[\"0042742\", \"defense response to bacterium\"], [\"0010150\", \"leaf senescence\"], [\"0009409\", \"response to cold\"], [\"0009753\", \"response to jasmonic acid\"], [\"0009416\", \"response to light stimulus\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cc40 @network_members=#<Interaction_Network:0x0000561fbd819150 @agi_locus1=\"At1g20020\", @agi_locus2=\"At4g22890\", @interactors=[[\"At1g60950\", \"At5g66190\"], \"direct\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0042742\", \"defense response to bacterium\"], [\"0050832\", \"defense response to fungus\"], [\"0015979\", \"photosynthesis\"], [\"0006260\", \"DNA replication\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0009643\", \"photosynthetic acclimation\"], [\"0009767\", \"photosynthetic electron transport chain\"], [\"0009853\", \"photorespiration\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0009959\", \"negative gravitropism\"], [\"0009639\", \"response to red or far red light\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cc18 @network_members=#<Interaction_Network:0x0000561fbd819128 @agi_locus1=\"At3g55800\", @agi_locus2=\"At4g22890\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0016051\", \"carbohydrate biosynthetic process\"], [\"0042742\", \"defense response to bacterium\"], [\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006002\", \"fructose 6-phosphate metabolic process\"], [\"0006000\", \"fructose metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0019252\", \"starch biosynthetic process\"], [\"0005986\", \"sucrose biosynthetic process\"], [\"0005975\", \"carbohydrate metabolic process\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cbf0 @network_members=#<Interaction_Network:0x0000561fbd8190b0 @agi_locus1=\"At3g28220\", @agi_locus2=\"At4g22890\", @interactors=[\"At4g20790\"]>, @kegg=[], @go=[[\"0009773\", \"photosynthetic electron transport in photosystem I\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cbc8 @network_members=#<Interaction_Network:0x0000561fbd819088 @agi_locus1=\"At3g62410\", @agi_locus2=\"At4g22890\", @interactors=[\"At5g42980\"]>, @kegg=[], @go=[[\"0071454\", \"cellular response to anoxia\"], [\"0070417\", \"cellular response to cold\"], [\"0034605\", \"cellular response to heat\"], [\"0080153\", \"negative regulation of reductive pentose-phosphate cycle\"], [\"0018316\", \"peptide cross-linking via L-cystine\"], [\"0065003\", \"protein-containing complex assembly\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cba0 @network_members=#<Interaction_Network:0x0000561fbd818fe8 @agi_locus1=\"At4g22890\", @agi_locus2=\"At5g35630\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00220\", \"Arginine biosynthesis\"], [\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01230\", \"Biosynthesis of amino acids\"]], @go=[[\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0007568\", \"aging\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006542\", \"glutamine biosynthetic process\"], [\"0046686\", \"response to cadmium ion\"], [\"0006807\", \"nitrogen compound metabolic process\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cb78 @network_members=#<Interaction_Network:0x0000561fbd818f70 @agi_locus1=\"At1g69870\", @agi_locus2=\"At4g22890\", @interactors=[\"At1g29060\", \"At5g37050\", \"At2g22425\", \"At3g12180\", \"At1g17280\", \"At4g37680\"]>, @kegg=[[\"ath04130\", \"SNARE interactions in vesicular transport\"], [\"ath03060\", \"Protein export\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"]], @go=[[\"0042128\", \"nitrate assimilation\"], [\"0015706\", \"nitrate transport\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"1901002\", \"positive regulation of response to salt stress\"], [\"0015031\", \"protein transport\"], [\"0047484\", \"regulation of response to osmotic stress\"], [\"0043182\", \"vacuolar sequestering of sodium ion\"], [\"0016192\", \"vesicle-mediated transport\"], [\"0045047\", \"protein targeting to ER\"], [\"0006465\", \"signal peptide processing\"], [\"0048658\", \"anther wall tapetum development\"], [\"0042631\", \"cellular response to water deprivation\"], [\"1902457\", \"negative regulation of stomatal opening\"], [\"0000209\", \"protein polyubiquitination\"], [\"0030433\", \"ubiquitin-dependent ERAD pathway\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"], [\"0009725\", \"response to hormone\"], [\"0009744\", \"response to sucrose\"], [\"0009738\", \"abscisic acid-activated signaling pathway\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cb50 @network_members=#<Interaction_Network:0x0000561fbd818f48 @agi_locus1=\"At4g22890\", @agi_locus2=\"At5g43750\", @interactors=[\"At1g29060\", \"At1g21240\", \"At5g37050\", \"At2g22425\", \"At3g12180\", \"At1g17280\", \"At5g59650\", \"At4g20790\"]>, @kegg=[[\"ath04130\", \"SNARE interactions in vesicular transport\"], [\"ath03060\", \"Protein export\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"]], @go=[[\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006979\", \"response to oxidative stress\"], [\"1901002\", \"positive regulation of response to salt stress\"], [\"0015031\", \"protein transport\"], [\"0047484\", \"regulation of response to osmotic stress\"], [\"0043182\", \"vacuolar sequestering of sodium ion\"], [\"0016192\", \"vesicle-mediated transport\"], [\"0007166\", \"cell surface receptor signaling pathway\"], [\"0006468\", \"protein phosphorylation\"], [\"0045047\", \"protein targeting to ER\"], [\"0006465\", \"signal peptide processing\"], [\"0048658\", \"anther wall tapetum development\"], [\"0042631\", \"cellular response to water deprivation\"], [\"1902457\", \"negative regulation of stomatal opening\"], [\"0000209\", \"protein polyubiquitination\"], [\"0030433\", \"ubiquitin-dependent ERAD pathway\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cb28 @network_members=#<Interaction_Network:0x0000561fbd818ea8 @agi_locus1=\"At3g50820\", @agi_locus2=\"At4g22890\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0010205\", \"photoinhibition\"], [\"0019684\", \"photosynthesis, light reaction\"], [\"0010207\", \"photosystem II assembly\"], [\"0042549\", \"photosystem II stabilization\"], [\"0035304\", \"regulation of protein dephosphorylation\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cb00 @network_members=#<Interaction_Network:0x0000561fbd818d90 @agi_locus1=\"At3g61260\", @agi_locus2=\"At4g22890\", @interactors=[\"At1g21240\", \"At5g37050\", \"At3g12180\"]>, @kegg=[], @go=[[\"0051607\", \"defense response to virus\"], [\"0051665\", \"membrane raft localization\"], [\"0009751\", \"response to salicylic acid\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0007166\", \"cell surface receptor signaling pathway\"], [\"0006468\", \"protein phosphorylation\"], [\"0016192\", \"vesicle-mediated transport\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cad8 @network_members=#<Interaction_Network:0x0000561fbd818d40 @agi_locus1=\"At4g22890\", @agi_locus2=\"At5g38990\", @interactors=[\"At1g29060\", \"At5g37050\", \"At3g12180\", \"At1g17280\", \"At5g59650\", \"At4g20790\"]>, @kegg=[[\"ath04130\", \"SNARE interactions in vesicular transport\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"]], @go=[[\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0046777\", \"protein autophosphorylation\"], [\"0010038\", \"response to metal ion\"], [\"1901002\", \"positive regulation of response to salt stress\"], [\"0015031\", \"protein transport\"], [\"0047484\", \"regulation of response to osmotic stress\"], [\"0043182\", \"vacuolar sequestering of sodium ion\"], [\"0016192\", \"vesicle-mediated transport\"], [\"0042631\", \"cellular response to water deprivation\"], [\"1902457\", \"negative regulation of stomatal opening\"], [\"0000209\", \"protein polyubiquitination\"], [\"0030433\", \"ubiquitin-dependent ERAD pathway\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0cab0 @network_members=#<Interaction_Network:0x0000561fbd818d18 @agi_locus1=\"At4g02770\", @agi_locus2=\"At4g22890\", @interactors=\"direct\">, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0140547\", \"acquisition of seed longevity\"], [\"0015979\", \"photosynthesis\"], [\"0035556\", \"intracellular signal transduction\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"]]>, #<Uso_General_Annotation:0x0000561fbcc0ca88 @network_members=#<Interaction_Network:0x0000561fbd818cf0 @agi_locus1=\"At1g37130\", @agi_locus2=\"At3g28220\", @interactors=[\"At3g20190\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0042128\", \"nitrate assimilation\"], [\"0006809\", \"nitric oxide biosynthetic process\"], [\"0009635\", \"response to herbicide\"], [\"0009416\", \"response to light stimulus\"], [\"0009610\", \"response to symbiotic fungus\"], [\"0080092\", \"regulation of pollen tube growth\"]]>, #<Uso_General_Annotation:0x0000561fbcc0ca60 @network_members=#<Interaction_Network:0x0000561fbd818ca0 @agi_locus1=\"At1g37130\", @agi_locus2=\"At2g18700\", @interactors=[\"At5g65430\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00500\", \"Starch and sucrose metabolism\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"]], @go=[[\"0042128\", \"nitrate assimilation\"], [\"0006809\", \"nitric oxide biosynthetic process\"], [\"0009635\", \"response to herbicide\"], [\"0009416\", \"response to light stimulus\"], [\"0009610\", \"response to symbiotic fungus\"], [\"0016311\", \"dephosphorylation\"], [\"0005992\", \"trehalose biosynthetic process\"], [\"0019222\", \"regulation of metabolic process\"], [\"0050826\", \"response to freezing\"], [\"0009742\", \"brassinosteroid mediated signaling pathway\"], [\"0009409\", \"response to cold\"]]>, #<Uso_General_Annotation:0x0000561fbcc0ca38 @network_members=#<Interaction_Network:0x0000561fbd818c78 @agi_locus1=\"At3g47340\", @agi_locus2=\"At5g65010\", @interactors=[\"At4g26840\"]>, @kegg=[[\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01230\", \"Biosynthesis of amino acids\"]], @go=[[\"0006529\", \"asparagine biosynthetic process\"], [\"0009063\", \"cellular amino acid catabolic process\"], [\"0043617\", \"cellular response to sucrose starvation\"], [\"0006541\", \"glutamine metabolic process\"], [\"0070981\", \"L-asparagine biosynthetic process\"], [\"0009646\", \"response to absence of light\"], [\"0009750\", \"response to fructose\"], [\"0009749\", \"response to glucose\"], [\"0009744\", \"response to sucrose\"], [\"0006096\", \"glycolytic process\"], [\"0097164\", \"ammonium ion metabolic process\"], [\"0042538\", \"hyperosmotic salinity response\"], [\"0010286\", \"heat acclimation\"], [\"0043433\", \"negative regulation of DNA-binding transcription factor activity\"], [\"0016925\", \"protein sumoylation\"], [\"0009408\", \"response to heat\"]]>, #<Uso_General_Annotation:0x0000561fbcc0ca10 @network_members=#<Interaction_Network:0x0000561fbd818c50 @agi_locus1=\"At3g01500\", @agi_locus2=\"At5g14740\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0015976\", \"carbon utilization\"], [\"0042742\", \"defense response to bacterium\"], [\"0050832\", \"defense response to fungus\"], [\"2000122\", \"negative regulation of stomatal complex development\"], [\"0015979\", \"photosynthesis\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0010037\", \"response to carbon dioxide\"], [\"0009409\", \"response to cold\"], [\"0006865\", \"amino acid transport\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c9e8 @network_members=#<Interaction_Network:0x0000561fbd818c28 @agi_locus1=\"At2g39730\", @agi_locus2=\"At3g01500\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0042742\", \"defense response to bacterium\"], [\"0010150\", \"leaf senescence\"], [\"0009409\", \"response to cold\"], [\"0009753\", \"response to jasmonic acid\"], [\"0009416\", \"response to light stimulus\"], [\"0015976\", \"carbon utilization\"], [\"0050832\", \"defense response to fungus\"], [\"2000122\", \"negative regulation of stomatal complex development\"], [\"0015979\", \"photosynthesis\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0010037\", \"response to carbon dioxide\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c9c0 @network_members=#<Interaction_Network:0x0000561fbd818bb0 @agi_locus1=\"At3g01500\", @agi_locus2=\"At3g55800\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0015976\", \"carbon utilization\"], [\"0042742\", \"defense response to bacterium\"], [\"0050832\", \"defense response to fungus\"], [\"2000122\", \"negative regulation of stomatal complex development\"], [\"0015979\", \"photosynthesis\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0010037\", \"response to carbon dioxide\"], [\"0009409\", \"response to cold\"], [\"0016051\", \"carbohydrate biosynthetic process\"], [\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006002\", \"fructose 6-phosphate metabolic process\"], [\"0006000\", \"fructose metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0019252\", \"starch biosynthetic process\"], [\"0005986\", \"sucrose biosynthetic process\"], [\"0005975\", \"carbohydrate metabolic process\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c998 @network_members=#<Interaction_Network:0x0000561fbd818b88 @agi_locus1=\"At3g01500\", @agi_locus2=\"At3g28220\", @interactors=[\"At2g28960\", \"At4g16620\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0015976\", \"carbon utilization\"], [\"0042742\", \"defense response to bacterium\"], [\"0050832\", \"defense response to fungus\"], [\"2000122\", \"negative regulation of stomatal complex development\"], [\"0015979\", \"photosynthesis\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0010037\", \"response to carbon dioxide\"], [\"0009409\", \"response to cold\"], [\"0048653\", \"anther development\"], [\"0006952\", \"defense response\"], [\"0007639\", \"homeostasis of number of meristem cells\"], [\"0006468\", \"protein phosphorylation\"], [\"0016032\", \"viral process\"], [\"0009738\", \"abscisic acid-activated signaling pathway\"], [\"0071485\", \"cellular response to absence of light\"], [\"0071244\", \"cellular response to carbon dioxide\"], [\"1901528\", \"hydrogen peroxide mediated signaling pathway involved in stomatal movement\"], [\"0009789\", \"positive regulation of abscisic acid-activated signaling pathway\"], [\"0090333\", \"regulation of stomatal closure\"], [\"0009737\", \"response to abscisic acid\"], [\"0009416\", \"response to light stimulus\"], [\"0007165\", \"signal transduction\"], [\"0030244\", \"cellulose biosynthetic process\"], [\"0071446\", \"cellular response to salicylic acid stimulus\"], [\"0031347\", \"regulation of defense response\"], [\"2000031\", \"regulation of salicylic acid mediated signaling pathway\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c970 @network_members=#<Interaction_Network:0x0000561fbd818b60 @agi_locus1=\"At3g01500\", @agi_locus2=\"At3g62410\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0015976\", \"carbon utilization\"], [\"0042742\", \"defense response to bacterium\"], [\"0050832\", \"defense response to fungus\"], [\"2000122\", \"negative regulation of stomatal complex development\"], [\"0015979\", \"photosynthesis\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0010037\", \"response to carbon dioxide\"], [\"0009409\", \"response to cold\"], [\"0071454\", \"cellular response to anoxia\"], [\"0070417\", \"cellular response to cold\"], [\"0034605\", \"cellular response to heat\"], [\"0080153\", \"negative regulation of reductive pentose-phosphate cycle\"], [\"0018316\", \"peptide cross-linking via L-cystine\"], [\"0065003\", \"protein-containing complex assembly\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c948 @network_members=#<Interaction_Network:0x0000561fbd818b10 @agi_locus1=\"At3g01500\", @agi_locus2=\"At5g35630\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00220\", \"Arginine biosynthesis\"], [\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"]], @go=[[\"0015976\", \"carbon utilization\"], [\"0042742\", \"defense response to bacterium\"], [\"0050832\", \"defense response to fungus\"], [\"2000122\", \"negative regulation of stomatal complex development\"], [\"0015979\", \"photosynthesis\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0010037\", \"response to carbon dioxide\"], [\"0009409\", \"response to cold\"], [\"0007568\", \"aging\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006542\", \"glutamine biosynthetic process\"], [\"0046686\", \"response to cadmium ion\"], [\"0006807\", \"nitrogen compound metabolic process\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c920 @network_members=#<Interaction_Network:0x0000561fbd818ae8 @agi_locus1=\"At3g01500\", @agi_locus2=\"At3g50820\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00195\", \"Photosynthesis\"]], @go=[[\"0015976\", \"carbon utilization\"], [\"0042742\", \"defense response to bacterium\"], [\"0050832\", \"defense response to fungus\"], [\"2000122\", \"negative regulation of stomatal complex development\"], [\"0015979\", \"photosynthesis\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0010037\", \"response to carbon dioxide\"], [\"0009409\", \"response to cold\"], [\"0010205\", \"photoinhibition\"], [\"0019684\", \"photosynthesis, light reaction\"], [\"0010207\", \"photosystem II assembly\"], [\"0042549\", \"photosystem II stabilization\"], [\"0035304\", \"regulation of protein dephosphorylation\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c8f8 @network_members=#<Interaction_Network:0x0000561fbd818ac0 @agi_locus1=\"At2g26080\", @agi_locus2=\"At4g33010\", @interactors=[\"At3g62870\"]>, @kegg=[[\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath03010\", \"Ribosome\"]], @go=[[\"0019464\", \"glycine decarboxylation via glycine cleavage system\"], [\"0006546\", \"glycine catabolic process\"], [\"0000398\", \"mRNA splicing, via spliceosome\"], [\"0046686\", \"response to cadmium ion\"], [\"0000470\", \"maturation of LSU-rRNA\"], [\"0042254\", \"ribosome biogenesis\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c8d0 @network_members=#<Interaction_Network:0x0000561fbd818a48 @agi_locus1=\"At3g54050\", @agi_locus2=\"At4g33010\", @interactors=[\"At3g62870\"]>, @kegg=[[\"ath00010\", \"Glycolysis / Gluconeogenesis\"], [\"ath00030\", \"Pentose phosphate pathway\"], [\"ath00051\", \"Fructose and mannose metabolism\"], [\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath03010\", \"Ribosome\"]], @go=[[\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006002\", \"fructose 6-phosphate metabolic process\"], [\"0006000\", \"fructose metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0015979\", \"photosynthesis\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0009409\", \"response to cold\"], [\"0005986\", \"sucrose biosynthetic process\"], [\"0005985\", \"sucrose metabolic process\"], [\"0005975\", \"carbohydrate metabolic process\"], [\"0006546\", \"glycine catabolic process\"], [\"0019464\", \"glycine decarboxylation via glycine cleavage system\"], [\"0046686\", \"response to cadmium ion\"], [\"0000470\", \"maturation of LSU-rRNA\"], [\"0042254\", \"ribosome biogenesis\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c8a8 @network_members=#<Interaction_Network:0x0000561fbd818a20 @agi_locus1=\"At1g68570\", @agi_locus2=\"At4g33010\", @interactors=[\"At3g62870\"]>, @kegg=[[\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath03010\", \"Ribosome\"]], @go=[[\"0010336\", \"gibberellic acid homeostasis\"], [\"0042128\", \"nitrate assimilation\"], [\"0006857\", \"oligopeptide transport\"], [\"0006546\", \"glycine catabolic process\"], [\"0019464\", \"glycine decarboxylation via glycine cleavage system\"], [\"0046686\", \"response to cadmium ion\"], [\"0000470\", \"maturation of LSU-rRNA\"], [\"0042254\", \"ribosome biogenesis\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c880 @network_members=#<Interaction_Network:0x0000561fbd8189f8 @agi_locus1=\"At2g17880\", @agi_locus2=\"At3g16250\", @interactors=[\"At3g48150\"]>, @kegg=[[\"ath04120\", \"Ubiquitin mediated proteolysis\"]], @go=[[\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0031145\", \"anaphase-promoting complex-dependent catabolic process\"], [\"0051301\", \"cell division\"], [\"0007091\", \"metaphase/anaphase transition of mitotic cell cycle\"], [\"0045842\", \"positive regulation of mitotic metaphase/anaphase transition\"], [\"0016567\", \"protein ubiquitination\"], [\"0031347\", \"regulation of defense response\"], [\"0007049\", \"cell cycle\"], [\"0030071\", \"regulation of mitotic metaphase/anaphase transition\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c858 @network_members=#<Interaction_Network:0x0000561fbd8189a8 @agi_locus1=\"At2g17880\", @agi_locus2=\"At4g02770\", @interactors=[\"At4g17680\", \"At3g48150\", \"At3g49580\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"]], @go=[[\"0140547\", \"acquisition of seed longevity\"], [\"0015979\", \"photosynthesis\"], [\"0035556\", \"intracellular signal transduction\"], [\"0006952\", \"defense response\"], [\"0009873\", \"ethylene-activated signaling pathway\"], [\"0009867\", \"jasmonic acid mediated signaling pathway\"], [\"0031347\", \"regulation of defense response\"], [\"2000022\", \"regulation of jasmonic acid mediated signaling pathway\"], [\"0009611\", \"response to wounding\"], [\"0031145\", \"anaphase-promoting complex-dependent catabolic process\"], [\"0051301\", \"cell division\"], [\"0007091\", \"metaphase/anaphase transition of mitotic cell cycle\"], [\"0045842\", \"positive regulation of mitotic metaphase/anaphase transition\"], [\"0016567\", \"protein ubiquitination\"], [\"0007049\", \"cell cycle\"], [\"0030071\", \"regulation of mitotic metaphase/anaphase transition\"], [\"0098869\", \"cellular oxidant detoxification\"], [\"0010438\", \"cellular response to sulfur starvation\"], [\"0009651\", \"response to salt stress\"], [\"0008654\", \"phospholipid biosynthetic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c830 @network_members=#<Interaction_Network:0x0000561fbd818980 @agi_locus1=\"At3g28220\", @agi_locus2=\"At3g49160\", @interactors=[\"At1g25320\"]>, @kegg=[[\"ath00010\", \"Glycolysis / Gluconeogenesis\"], [\"ath00230\", \"Purine metabolism\"], [\"ath00620\", \"Pyruvate metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"], [\"ath01240\", \"Biosynthesis of cofactors\"]], @go=[[\"0006096\", \"glycolytic process\"], [\"0051321\", \"meiotic cell cycle\"], [\"0007064\", \"mitotic sister chromatid cohesion\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c808 @network_members=#<Interaction_Network:0x0000561fbd818958 @agi_locus1=\"At4g21210\", @agi_locus2=\"At4g37800\", @interactors=[\"At4g02930\"]>, @kegg=[[\"ath04626\", \"Plant-pathogen interaction\"]], @go=[[\"0042546\", \"cell wall biogenesis\"], [\"0071555\", \"cell wall organization\"], [\"0010411\", \"xyloglucan metabolic process\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0070125\", \"mitochondrial translational elongation\"], [\"0046686\", \"response to cadmium ion\"], [\"0006414\", \"translational elongation\"], [\"0042545\", \"cell wall modification\"], [\"0045490\", \"pectin catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c7e0 @network_members=#<Interaction_Network:0x0000561fbd8188e0 @agi_locus1=\"At2g26740\", @agi_locus2=\"At4g21210\", @interactors=[\"At4g02930\"]>, @kegg=[[\"ath04626\", \"Plant-pathogen interaction\"]], @go=[[\"0009733\", \"response to auxin\"], [\"0009414\", \"response to water deprivation\"], [\"0070125\", \"mitochondrial translational elongation\"], [\"0046686\", \"response to cadmium ion\"], [\"0006414\", \"translational elongation\"], [\"0042545\", \"cell wall modification\"], [\"0045490\", \"pectin catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c7b8 @network_members=#<Interaction_Network:0x0000561fbd8188b8 @agi_locus1=\"At2g39730\", @agi_locus2=\"At5g14740\", @interactors=[\"At5g42980\", \"At2g23070\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath03008\", \"Ribosome biogenesis in eukaryotes\"], [\"ath04712\", \"Circadian rhythm - plant\"]], @go=[[\"0042742\", \"defense response to bacterium\"], [\"0010150\", \"leaf senescence\"], [\"0009409\", \"response to cold\"], [\"0009753\", \"response to jasmonic acid\"], [\"0009416\", \"response to light stimulus\"], [\"0015976\", \"carbon utilization\"], [\"0006865\", \"amino acid transport\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"], [\"0071215\", \"cellular response to abscisic acid stimulus\"], [\"0010019\", \"chloroplast-nucleus signaling pathway\"], [\"0010187\", \"negative regulation of seed germination\"], [\"0018105\", \"peptidyl-serine phosphorylation\"], [\"0018107\", \"peptidyl-threonine phosphorylation\"], [\"0051726\", \"regulation of cell cycle\"], [\"0040008\", \"regulation of growth\"], [\"2000028\", \"regulation of photoperiodism, flowering\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c790 @network_members=#<Interaction_Network:0x0000561fbd818890 @agi_locus1=\"At3g55800\", @agi_locus2=\"At5g14740\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"]], @go=[[\"0016051\", \"carbohydrate biosynthetic process\"], [\"0042742\", \"defense response to bacterium\"], [\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006002\", \"fructose 6-phosphate metabolic process\"], [\"0006000\", \"fructose metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0019252\", \"starch biosynthetic process\"], [\"0005986\", \"sucrose biosynthetic process\"], [\"0005975\", \"carbohydrate metabolic process\"], [\"0015976\", \"carbon utilization\"], [\"0006865\", \"amino acid transport\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c768 @network_members=#<Interaction_Network:0x0000561fbd818840 @agi_locus1=\"At2g18700\", @agi_locus2=\"At5g14740\", @interactors=[\"At2g23070\"]>, @kegg=[[\"ath00500\", \"Starch and sucrose metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath03008\", \"Ribosome biogenesis in eukaryotes\"], [\"ath04712\", \"Circadian rhythm - plant\"]], @go=[[\"0016311\", \"dephosphorylation\"], [\"0005992\", \"trehalose biosynthetic process\"], [\"0015976\", \"carbon utilization\"], [\"0042742\", \"defense response to bacterium\"], [\"0006865\", \"amino acid transport\"], [\"0071215\", \"cellular response to abscisic acid stimulus\"], [\"0010019\", \"chloroplast-nucleus signaling pathway\"], [\"0010187\", \"negative regulation of seed germination\"], [\"0018105\", \"peptidyl-serine phosphorylation\"], [\"0018107\", \"peptidyl-threonine phosphorylation\"], [\"0051726\", \"regulation of cell cycle\"], [\"0040008\", \"regulation of growth\"], [\"2000028\", \"regulation of photoperiodism, flowering\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c740 @network_members=#<Interaction_Network:0x0000561fbd818818 @agi_locus1=\"At3g62410\", @agi_locus2=\"At5g14740\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0071454\", \"cellular response to anoxia\"], [\"0070417\", \"cellular response to cold\"], [\"0034605\", \"cellular response to heat\"], [\"0080153\", \"negative regulation of reductive pentose-phosphate cycle\"], [\"0018316\", \"peptide cross-linking via L-cystine\"], [\"0065003\", \"protein-containing complex assembly\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0015976\", \"carbon utilization\"], [\"0042742\", \"defense response to bacterium\"], [\"0006865\", \"amino acid transport\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c718 @network_members=#<Interaction_Network:0x0000561fbd8187f0 @agi_locus1=\"At5g14740\", @agi_locus2=\"At5g35630\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00220\", \"Arginine biosynthesis\"], [\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"]], @go=[[\"0015976\", \"carbon utilization\"], [\"0042742\", \"defense response to bacterium\"], [\"0006865\", \"amino acid transport\"], [\"0007568\", \"aging\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006542\", \"glutamine biosynthetic process\"], [\"0046686\", \"response to cadmium ion\"], [\"0006807\", \"nitrogen compound metabolic process\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c6f0 @network_members=#<Interaction_Network:0x0000561fbd818778 @agi_locus1=\"At3g50820\", @agi_locus2=\"At5g14740\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00910\", \"Nitrogen metabolism\"]], @go=[[\"0010205\", \"photoinhibition\"], [\"0019684\", \"photosynthesis, light reaction\"], [\"0010207\", \"photosystem II assembly\"], [\"0042549\", \"photosystem II stabilization\"], [\"0035304\", \"regulation of protein dephosphorylation\"], [\"0015976\", \"carbon utilization\"], [\"0042742\", \"defense response to bacterium\"], [\"0006865\", \"amino acid transport\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c6c8 @network_members=#<Interaction_Network:0x0000561fbd818750 @agi_locus1=\"At2g34420\", @agi_locus2=\"At4g37980\", @interactors=[\"At4g34840\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00940\", \"Phenylpropanoid biosynthesis\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath00270\", \"Cysteine and methionine metabolism\"]], @go=[[\"0009765\", \"photosynthesis, light harvesting\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009416\", \"response to light stimulus\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0010597\", \"green leaf volatile biosynthetic process\"], [\"0009809\", \"lignin biosynthetic process\"], [\"0009617\", \"response to bacterium\"], [\"0033512\", \"L-lysine catabolic process to acetyl-CoA via saccharopine\"], [\"0019509\", \"L-methionine salvage from methylthioadenosine\"], [\"0009116\", \"nucleoside metabolic process\"], [\"0010087\", \"phloem or xylem histogenesis\"], [\"0000003\", \"reproduction\"], [\"0009627\", \"systemic acquired resistance\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c6a0 @network_members=#<Interaction_Network:0x0000561fbd818728 @agi_locus1=\"At3g27690\", @agi_locus2=\"At4g37980\", @interactors=[\"At4g34840\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00940\", \"Phenylpropanoid biosynthesis\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath00270\", \"Cysteine and methionine metabolism\"]], @go=[[\"0045893\", \"positive regulation of transcription, DNA-templated\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0071215\", \"cellular response to abscisic acid stimulus\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"1903428\", \"positive regulation of reactive oxygen species biosynthetic process\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0090333\", \"regulation of stomatal closure\"], [\"0009409\", \"response to cold\"], [\"0009269\", \"response to desiccation\"], [\"0010218\", \"response to far red light\"], [\"0009644\", \"response to high light intensity\"], [\"0009416\", \"response to light stimulus\"], [\"0009645\", \"response to low light intensity stimulus\"], [\"0010114\", \"response to red light\"], [\"0030104\", \"water homeostasis\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0010597\", \"green leaf volatile biosynthetic process\"], [\"0009809\", \"lignin biosynthetic process\"], [\"0009617\", \"response to bacterium\"], [\"0033512\", \"L-lysine catabolic process to acetyl-CoA via saccharopine\"], [\"0019509\", \"L-methionine salvage from methylthioadenosine\"], [\"0009116\", \"nucleoside metabolic process\"], [\"0010087\", \"phloem or xylem histogenesis\"], [\"0000003\", \"reproduction\"], [\"0009627\", \"systemic acquired resistance\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c678 @network_members=#<Interaction_Network:0x0000561fbd8186d8 @agi_locus1=\"At1g61520\", @agi_locus2=\"At4g37980\", @interactors=[\"At2g16740\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00940\", \"Phenylpropanoid biosynthesis\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"]], @go=[[\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009409\", \"response to cold\"], [\"0009644\", \"response to high light intensity\"], [\"0009416\", \"response to light stimulus\"], [\"0009645\", \"response to low light intensity stimulus\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0010597\", \"green leaf volatile biosynthetic process\"], [\"0009809\", \"lignin biosynthetic process\"], [\"0009617\", \"response to bacterium\"], [\"0033512\", \"L-lysine catabolic process to acetyl-CoA via saccharopine\"], [\"0000209\", \"protein polyubiquitination\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c650 @network_members=#<Interaction_Network:0x0000561fbd8186b0 @agi_locus1=\"At2g39730\", @agi_locus2=\"At3g55800\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0042742\", \"defense response to bacterium\"], [\"0010150\", \"leaf senescence\"], [\"0009409\", \"response to cold\"], [\"0009753\", \"response to jasmonic acid\"], [\"0009416\", \"response to light stimulus\"], [\"0016051\", \"carbohydrate biosynthetic process\"], [\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006002\", \"fructose 6-phosphate metabolic process\"], [\"0006000\", \"fructose metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0019252\", \"starch biosynthetic process\"], [\"0005986\", \"sucrose biosynthetic process\"], [\"0005975\", \"carbohydrate metabolic process\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c628 @network_members=#<Interaction_Network:0x0000561fbd818688 @agi_locus1=\"At2g18700\", @agi_locus2=\"At2g39730\", @interactors=[\"At2g23070\"]>, @kegg=[[\"ath00500\", \"Starch and sucrose metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath03008\", \"Ribosome biogenesis in eukaryotes\"], [\"ath04712\", \"Circadian rhythm - plant\"]], @go=[[\"0016311\", \"dephosphorylation\"], [\"0005992\", \"trehalose biosynthetic process\"], [\"0042742\", \"defense response to bacterium\"], [\"0010150\", \"leaf senescence\"], [\"0009409\", \"response to cold\"], [\"0009753\", \"response to jasmonic acid\"], [\"0009416\", \"response to light stimulus\"], [\"0071215\", \"cellular response to abscisic acid stimulus\"], [\"0010019\", \"chloroplast-nucleus signaling pathway\"], [\"0010187\", \"negative regulation of seed germination\"], [\"0018105\", \"peptidyl-serine phosphorylation\"], [\"0018107\", \"peptidyl-threonine phosphorylation\"], [\"0051726\", \"regulation of cell cycle\"], [\"0040008\", \"regulation of growth\"], [\"2000028\", \"regulation of photoperiodism, flowering\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c600 @network_members=#<Interaction_Network:0x0000561fbd818610 @agi_locus1=\"At2g39730\", @agi_locus2=\"At3g62410\", @interactors=[\"At5g42980\"]>, @kegg=[], @go=[[\"0042742\", \"defense response to bacterium\"], [\"0010150\", \"leaf senescence\"], [\"0009409\", \"response to cold\"], [\"0009753\", \"response to jasmonic acid\"], [\"0009416\", \"response to light stimulus\"], [\"0071454\", \"cellular response to anoxia\"], [\"0070417\", \"cellular response to cold\"], [\"0034605\", \"cellular response to heat\"], [\"0080153\", \"negative regulation of reductive pentose-phosphate cycle\"], [\"0018316\", \"peptide cross-linking via L-cystine\"], [\"0065003\", \"protein-containing complex assembly\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0009744\", \"response to sucrose\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c5d8 @network_members=#<Interaction_Network:0x0000561fbd8185e8 @agi_locus1=\"At2g39730\", @agi_locus2=\"At5g35630\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00220\", \"Arginine biosynthesis\"], [\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01230\", \"Biosynthesis of amino acids\"]], @go=[[\"0042742\", \"defense response to bacterium\"], [\"0010150\", \"leaf senescence\"], [\"0009409\", \"response to cold\"], [\"0009753\", \"response to jasmonic acid\"], [\"0009416\", \"response to light stimulus\"], [\"0007568\", \"aging\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006542\", \"glutamine biosynthetic process\"], [\"0046686\", \"response to cadmium ion\"], [\"0006807\", \"nitrogen compound metabolic process\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c5b0 @network_members=#<Interaction_Network:0x0000561fbd8185c0 @agi_locus1=\"At2g39730\", @agi_locus2=\"At3g50820\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0042742\", \"defense response to bacterium\"], [\"0010150\", \"leaf senescence\"], [\"0009409\", \"response to cold\"], [\"0009753\", \"response to jasmonic acid\"], [\"0009416\", \"response to light stimulus\"], [\"0010205\", \"photoinhibition\"], [\"0019684\", \"photosynthesis, light reaction\"], [\"0010207\", \"photosystem II assembly\"], [\"0042549\", \"photosystem II stabilization\"], [\"0035304\", \"regulation of protein dephosphorylation\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c588 @network_members=#<Interaction_Network:0x0000561fbd818548 @agi_locus1=\"At1g20020\", @agi_locus2=\"At4g02770\", @interactors=[\"At4g22890\", \"At4g11960\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0042742\", \"defense response to bacterium\"], [\"0050832\", \"defense response to fungus\"], [\"0015979\", \"photosynthesis\"], [\"0006260\", \"DNA replication\"], [\"0140547\", \"acquisition of seed longevity\"], [\"0035556\", \"intracellular signal transduction\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c560 @network_members=#<Interaction_Network:0x0000561fbd8184f8 @agi_locus1=\"At2g34420\", @agi_locus2=\"At5g54270\", @interactors=[\"At4g37140\", \"At3g44680\", \"At4g34840\", \"At1g79870\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00270\", \"Cysteine and methionine metabolism\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0009765\", \"photosynthesis, light harvesting\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009416\", \"response to light stimulus\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0015979\", \"photosynthesis\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0009737\", \"response to abscisic acid\"], [\"0009635\", \"response to herbicide\"], [\"0009644\", \"response to high light intensity\"], [\"0006412\", \"translation\"], [\"0009793\", \"embryo development ending in seed dormancy\"], [\"1990619\", \"histone H3-K9 deacetylation\"], [\"0010187\", \"negative regulation of seed germination\"], [\"1900055\", \"regulation of leaf senescence\"], [\"0050832\", \"defense response to fungus\"], [\"0009626\", \"plant-type hypersensitive response\"], [\"0019509\", \"L-methionine salvage from methylthioadenosine\"], [\"0009116\", \"nucleoside metabolic process\"], [\"0010087\", \"phloem or xylem histogenesis\"], [\"0000003\", \"reproduction\"], [\"0009627\", \"systemic acquired resistance\"], [\"0009854\", \"oxidative photosynthetic carbon pathway\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c538 @network_members=#<Interaction_Network:0x0000561fbd8184d0 @agi_locus1=\"At2g34420\", @agi_locus2=\"At3g27690\", @interactors=[\"At3g57870\", \"At4g26220\", \"At3g48040\", \"At4g34840\", \"At1g79870\", \"At5g50240\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath03013\", \"Nucleocytoplasmic transport\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath00940\", \"Phenylpropanoid biosynthesis\"], [\"ath00941\", \"Flavonoid biosynthesis\"], [\"ath00945\", \"Stilbenoid, diarylheptanoid and gingerol biosynthesis\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath04145\", \"Phagosome\"], [\"ath00270\", \"Cysteine and methionine metabolism\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0009765\", \"photosynthesis, light harvesting\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009416\", \"response to light stimulus\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0045893\", \"positive regulation of transcription, DNA-templated\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0071215\", \"cellular response to abscisic acid stimulus\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"1903428\", \"positive regulation of reactive oxygen species biosynthetic process\"], [\"0090333\", \"regulation of stomatal closure\"], [\"0009409\", \"response to cold\"], [\"0009269\", \"response to desiccation\"], [\"0010218\", \"response to far red light\"], [\"0009644\", \"response to high light intensity\"], [\"0009645\", \"response to low light intensity stimulus\"], [\"0010114\", \"response to red light\"], [\"0030104\", \"water homeostasis\"], [\"0009793\", \"embryo development ending in seed dormancy\"], [\"0016925\", \"protein sumoylation\"], [\"0009737\", \"response to abscisic acid\"], [\"0009809\", \"lignin biosynthetic process\"], [\"0032259\", \"methylation\"], [\"0005992\", \"trehalose biosynthetic process\"], [\"0015031\", \"protein transport\"], [\"0009738\", \"abscisic acid-activated signaling pathway\"], [\"0007015\", \"actin filament organization\"], [\"0030865\", \"cortical cytoskeleton organization\"], [\"0007163\", \"establishment or maintenance of cell polarity\"], [\"0009788\", \"negative regulation of abscisic acid-activated signaling pathway\"], [\"0032956\", \"regulation of actin cytoskeleton organization\"], [\"0008360\", \"regulation of cell shape\"], [\"0007264\", \"small GTPase mediated signal transduction\"], [\"0019509\", \"L-methionine salvage from methylthioadenosine\"], [\"0009116\", \"nucleoside metabolic process\"], [\"0010087\", \"phloem or xylem histogenesis\"], [\"0000003\", \"reproduction\"], [\"0009627\", \"systemic acquired resistance\"], [\"0009854\", \"oxidative photosynthetic carbon pathway\"], [\"0030091\", \"protein repair\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c510 @network_members=#<Interaction_Network:0x0000561fbd818458 @agi_locus1=\"At2g34420\", @agi_locus2=\"At3g16250\", @interactors=[\"At3g47620\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0009765\", \"photosynthesis, light harvesting\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009416\", \"response to light stimulus\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0010229\", \"inflorescence development\"], [\"0031347\", \"regulation of defense response\"], [\"0010029\", \"regulation of seed germination\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c4e8 @network_members=#<Interaction_Network:0x0000561fbd818430 @agi_locus1=\"At2g34420\", @agi_locus2=\"At5g35630\", @interactors=[\"At1g79870\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00220\", \"Arginine biosynthesis\"], [\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0009765\", \"photosynthesis, light harvesting\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009416\", \"response to light stimulus\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0007568\", \"aging\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006542\", \"glutamine biosynthetic process\"], [\"0046686\", \"response to cadmium ion\"], [\"0006807\", \"nitrogen compound metabolic process\"], [\"0009854\", \"oxidative photosynthetic carbon pathway\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c4c0 @network_members=#<Interaction_Network:0x0000561fbd818390 @agi_locus1=\"At1g29910\", @agi_locus2=\"At2g34420\", @interactors=[\"At3g47620\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009750\", \"response to fructose\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0009733\", \"response to auxin\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0010229\", \"inflorescence development\"], [\"0031347\", \"regulation of defense response\"], [\"0010029\", \"regulation of seed germination\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c498 @network_members=#<Interaction_Network:0x0000561fbd8182c8 @agi_locus1=\"At2g34420\", @agi_locus2=\"At3g61260\", @interactors=[\"At3g48040\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath04145\", \"Phagosome\"]], @go=[[\"0009765\", \"photosynthesis, light harvesting\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009416\", \"response to light stimulus\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0051607\", \"defense response to virus\"], [\"0051665\", \"membrane raft localization\"], [\"0009751\", \"response to salicylic acid\"], [\"0009738\", \"abscisic acid-activated signaling pathway\"], [\"0007015\", \"actin filament organization\"], [\"0030865\", \"cortical cytoskeleton organization\"], [\"0007163\", \"establishment or maintenance of cell polarity\"], [\"0009788\", \"negative regulation of abscisic acid-activated signaling pathway\"], [\"0032956\", \"regulation of actin cytoskeleton organization\"], [\"0008360\", \"regulation of cell shape\"], [\"0007264\", \"small GTPase mediated signal transduction\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c470 @network_members=#<Interaction_Network:0x0000561fbd8182a0 @agi_locus1=\"At1g29920\", @agi_locus2=\"At2g34420\", @interactors=[\"At3g47620\", \"At3g26090\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009750\", \"response to fructose\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0010229\", \"inflorescence development\"], [\"0031347\", \"regulation of defense response\"], [\"0010029\", \"regulation of seed germination\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"], [\"0009867\", \"jasmonic acid mediated signaling pathway\"], [\"0009968\", \"negative regulation of signal transduction\"], [\"0043547\", \"positive regulation of GTPase activity\"], [\"0042127\", \"regulation of cell population proliferation\"], [\"0009749\", \"response to glucose\"], [\"0009414\", \"response to water deprivation\"], [\"0010182\", \"sugar mediated signaling pathway\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c448 @network_members=#<Interaction_Network:0x0000561fbd818278 @agi_locus1=\"At2g34420\", @agi_locus2=\"At4g02770\", @interactors=[\"At3g47620\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00195\", \"Photosynthesis\"]], @go=[[\"0009765\", \"photosynthesis, light harvesting\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009416\", \"response to light stimulus\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0140547\", \"acquisition of seed longevity\"], [\"0015979\", \"photosynthesis\"], [\"0035556\", \"intracellular signal transduction\"], [\"0010229\", \"inflorescence development\"], [\"0031347\", \"regulation of defense response\"], [\"0010029\", \"regulation of seed germination\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c420 @network_members=#<Interaction_Network:0x0000561fbd818188 @agi_locus1=\"At2g34420\", @agi_locus2=\"At5g01530\", @interactors=[\"At3g26090\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0009765\", \"photosynthesis, light harvesting\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009416\", \"response to light stimulus\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0009867\", \"jasmonic acid mediated signaling pathway\"], [\"0009968\", \"negative regulation of signal transduction\"], [\"0043547\", \"positive regulation of GTPase activity\"], [\"0042127\", \"regulation of cell population proliferation\"], [\"0009737\", \"response to abscisic acid\"], [\"0009749\", \"response to glucose\"], [\"0009414\", \"response to water deprivation\"], [\"0010182\", \"sugar mediated signaling pathway\"], [\"0006355\", \"regulation of transcription, DNA-templated\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c3f8 @network_members=#<Interaction_Network:0x0000561fbd8180e8 @agi_locus1=\"At2g26740\", @agi_locus2=\"At4g37800\", @interactors=[\"At4g02930\"]>, @kegg=[[\"ath04626\", \"Plant-pathogen interaction\"]], @go=[[\"0009733\", \"response to auxin\"], [\"0009414\", \"response to water deprivation\"], [\"0042546\", \"cell wall biogenesis\"], [\"0071555\", \"cell wall organization\"], [\"0010411\", \"xyloglucan metabolic process\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0070125\", \"mitochondrial translational elongation\"], [\"0046686\", \"response to cadmium ion\"], [\"0006414\", \"translational elongation\"], [\"0042545\", \"cell wall modification\"], [\"0045490\", \"pectin catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c3d0 @network_members=#<Interaction_Network:0x0000561fbd8180c0 @agi_locus1=\"At3g47340\", @agi_locus2=\"At4g37800\", @interactors=[\"At4g37550\"]>, @kegg=[[\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01230\", \"Biosynthesis of amino acids\"], [\"ath00460\", \"Cyanoamino acid metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0006529\", \"asparagine biosynthetic process\"], [\"0009063\", \"cellular amino acid catabolic process\"], [\"0043617\", \"cellular response to sucrose starvation\"], [\"0006541\", \"glutamine metabolic process\"], [\"0070981\", \"L-asparagine biosynthetic process\"], [\"0009646\", \"response to absence of light\"], [\"0009750\", \"response to fructose\"], [\"0009749\", \"response to glucose\"], [\"0009744\", \"response to sucrose\"], [\"0006096\", \"glycolytic process\"], [\"0042546\", \"cell wall biogenesis\"], [\"0071555\", \"cell wall organization\"], [\"0010411\", \"xyloglucan metabolic process\"], [\"0006355\", \"regulation of transcription, DNA-templated\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c3a8 @network_members=#<Interaction_Network:0x0000561fbd818070 @agi_locus1=\"At1g12900\", @agi_locus2=\"At1g61520\", @interactors=[\"At5g19320\"]>, @kegg=[[\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath03013\", \"Nucleocytoplasmic transport\"]], @go=[[\"0006006\", \"glucose metabolic process\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0008643\", \"carbohydrate transport\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009409\", \"response to cold\"], [\"0009644\", \"response to high light intensity\"], [\"0009416\", \"response to light stimulus\"], [\"0009645\", \"response to low light intensity stimulus\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0006913\", \"nucleocytoplasmic transport\"], [\"0043547\", \"positive regulation of GTPase activity\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c380 @network_members=#<Interaction_Network:0x0000561fbd81bbf8 @agi_locus1=\"At1g12900\", @agi_locus2=\"At3g45780\", @interactors=[\"At1g44820\", \"At3g54670\"]>, @kegg=[[\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath00220\", \"Arginine biosynthesis\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01210\", \"2-Oxocarboxylic acid metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"]], @go=[[\"0006006\", \"glucose metabolic process\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0008643\", \"carbohydrate transport\"], [\"0009904\", \"chloroplast accumulation movement\"], [\"0009903\", \"chloroplast avoidance movement\"], [\"0007623\", \"circadian rhythm\"], [\"0010362\", \"negative regulation of anion channel activity by blue light\"], [\"0009638\", \"phototropism\"], [\"0046777\", \"protein autophosphorylation\"], [\"0006468\", \"protein phosphorylation\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0010155\", \"regulation of proton transport\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0009637\", \"response to blue light\"], [\"0050896\", \"response to stimulus\"], [\"0006520\", \"cellular amino acid metabolic process\"], [\"0051301\", \"cell division\"], [\"0007059\", \"chromosome segregation\"], [\"0006281\", \"DNA repair\"], [\"0051321\", \"meiotic cell cycle\"], [\"0007064\", \"mitotic sister chromatid cohesion\"], [\"0007062\", \"sister chromatid cohesion\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c358 @network_members=#<Interaction_Network:0x0000561fbd81b8d8 @agi_locus1=\"At3g55800\", @agi_locus2=\"At3g62410\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0016051\", \"carbohydrate biosynthetic process\"], [\"0042742\", \"defense response to bacterium\"], [\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006002\", \"fructose 6-phosphate metabolic process\"], [\"0006000\", \"fructose metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0019252\", \"starch biosynthetic process\"], [\"0005986\", \"sucrose biosynthetic process\"], [\"0005975\", \"carbohydrate metabolic process\"], [\"0071454\", \"cellular response to anoxia\"], [\"0070417\", \"cellular response to cold\"], [\"0034605\", \"cellular response to heat\"], [\"0080153\", \"negative regulation of reductive pentose-phosphate cycle\"], [\"0018316\", \"peptide cross-linking via L-cystine\"], [\"0065003\", \"protein-containing complex assembly\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c330 @network_members=#<Interaction_Network:0x0000561fbd81b680 @agi_locus1=\"At3g55800\", @agi_locus2=\"At5g35630\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath00220\", \"Arginine biosynthesis\"], [\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"]], @go=[[\"0016051\", \"carbohydrate biosynthetic process\"], [\"0042742\", \"defense response to bacterium\"], [\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006002\", \"fructose 6-phosphate metabolic process\"], [\"0006000\", \"fructose metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0019252\", \"starch biosynthetic process\"], [\"0005986\", \"sucrose biosynthetic process\"], [\"0005975\", \"carbohydrate metabolic process\"], [\"0007568\", \"aging\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006542\", \"glutamine biosynthetic process\"], [\"0046686\", \"response to cadmium ion\"], [\"0006807\", \"nitrogen compound metabolic process\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c308 @network_members=#<Interaction_Network:0x0000561fbd81b3d8 @agi_locus1=\"At3g50820\", @agi_locus2=\"At3g55800\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0010205\", \"photoinhibition\"], [\"0019684\", \"photosynthesis, light reaction\"], [\"0010207\", \"photosystem II assembly\"], [\"0042549\", \"photosystem II stabilization\"], [\"0035304\", \"regulation of protein dephosphorylation\"], [\"0016051\", \"carbohydrate biosynthetic process\"], [\"0042742\", \"defense response to bacterium\"], [\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006002\", \"fructose 6-phosphate metabolic process\"], [\"0006000\", \"fructose metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0019252\", \"starch biosynthetic process\"], [\"0005986\", \"sucrose biosynthetic process\"], [\"0005975\", \"carbohydrate metabolic process\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c2e0 @network_members=#<Interaction_Network:0x0000561fbd81b158 @agi_locus1=\"At3g45780\", @agi_locus2=\"At5g17300\", @interactors=[\"At1g10210\"]>, @kegg=[[\"ath04016\", \"MAPK signaling pathway - plant\"]], @go=[[\"0009904\", \"chloroplast accumulation movement\"], [\"0009903\", \"chloroplast avoidance movement\"], [\"0007623\", \"circadian rhythm\"], [\"0010362\", \"negative regulation of anion channel activity by blue light\"], [\"0009638\", \"phototropism\"], [\"0046777\", \"protein autophosphorylation\"], [\"0006468\", \"protein phosphorylation\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0010155\", \"regulation of proton transport\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0009637\", \"response to blue light\"], [\"0050896\", \"response to stimulus\"], [\"0009734\", \"auxin-activated signaling pathway\"], [\"0010600\", \"regulation of auxin biosynthetic process\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0016567\", \"protein ubiquitination\"], [\"0035556\", \"intracellular signal transduction\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c2b8 @network_members=#<Interaction_Network:0x0000561fbd81aeb0 @agi_locus1=\"At2g34420\", @agi_locus2=\"At3g27690\", @interactors=[\"At3g57870\", \"At4g26220\", \"At5g50240\", \"At1g79870\", \"At3g48040\", \"At4g34840\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath03013\", \"Nucleocytoplasmic transport\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath00940\", \"Phenylpropanoid biosynthesis\"], [\"ath00941\", \"Flavonoid biosynthesis\"], [\"ath00945\", \"Stilbenoid, diarylheptanoid and gingerol biosynthesis\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath04145\", \"Phagosome\"], [\"ath00270\", \"Cysteine and methionine metabolism\"]], @go=[[\"0009765\", \"photosynthesis, light harvesting\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009416\", \"response to light stimulus\"], [\"0032875\", \"regulation of DNA endoreduplication\"], [\"0045893\", \"positive regulation of transcription, DNA-templated\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0071215\", \"cellular response to abscisic acid stimulus\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"1903428\", \"positive regulation of reactive oxygen species biosynthetic process\"], [\"0090333\", \"regulation of stomatal closure\"], [\"0009409\", \"response to cold\"], [\"0009269\", \"response to desiccation\"], [\"0010218\", \"response to far red light\"], [\"0009644\", \"response to high light intensity\"], [\"0009645\", \"response to low light intensity stimulus\"], [\"0010114\", \"response to red light\"], [\"0030104\", \"water homeostasis\"], [\"0009793\", \"embryo development ending in seed dormancy\"], [\"0016925\", \"protein sumoylation\"], [\"0009737\", \"response to abscisic acid\"], [\"0009809\", \"lignin biosynthetic process\"], [\"0032259\", \"methylation\"], [\"0005992\", \"trehalose biosynthetic process\"], [\"0015031\", \"protein transport\"], [\"0030091\", \"protein repair\"], [\"0009854\", \"oxidative photosynthetic carbon pathway\"], [\"0009738\", \"abscisic acid-activated signaling pathway\"], [\"0007015\", \"actin filament organization\"], [\"0030865\", \"cortical cytoskeleton organization\"], [\"0007163\", \"establishment or maintenance of cell polarity\"], [\"0009788\", \"negative regulation of abscisic acid-activated signaling pathway\"], [\"0032956\", \"regulation of actin cytoskeleton organization\"], [\"0008360\", \"regulation of cell shape\"], [\"0007264\", \"small GTPase mediated signal transduction\"], [\"0019509\", \"L-methionine salvage from methylthioadenosine\"], [\"0009116\", \"nucleoside metabolic process\"], [\"0010087\", \"phloem or xylem histogenesis\"], [\"0000003\", \"reproduction\"], [\"0009627\", \"systemic acquired resistance\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c290 @network_members=#<Interaction_Network:0x0000561fbd81ac80 @agi_locus1=\"At3g27690\", @agi_locus2=\"At5g35630\", @interactors=[\"At1g79870\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00220\", \"Arginine biosynthesis\"], [\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0045893\", \"positive regulation of transcription, DNA-templated\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0071215\", \"cellular response to abscisic acid stimulus\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"1903428\", \"positive regulation of reactive oxygen species biosynthetic process\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0090333\", \"regulation of stomatal closure\"], [\"0009409\", \"response to cold\"], [\"0009269\", \"response to desiccation\"], [\"0010218\", \"response to far red light\"], [\"0009644\", \"response to high light intensity\"], [\"0009416\", \"response to light stimulus\"], [\"0009645\", \"response to low light intensity stimulus\"], [\"0010114\", \"response to red light\"], [\"0030104\", \"water homeostasis\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0007568\", \"aging\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006542\", \"glutamine biosynthetic process\"], [\"0046686\", \"response to cadmium ion\"], [\"0006807\", \"nitrogen compound metabolic process\"], [\"0009854\", \"oxidative photosynthetic carbon pathway\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c268 @network_members=#<Interaction_Network:0x0000561fbd81aaa0 @agi_locus1=\"At3g27690\", @agi_locus2=\"At3g61260\", @interactors=[\"At3g48040\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath04145\", \"Phagosome\"]], @go=[[\"0045893\", \"positive regulation of transcription, DNA-templated\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0071215\", \"cellular response to abscisic acid stimulus\"], [\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0009769\", \"photosynthesis, light harvesting in photosystem II\"], [\"1903428\", \"positive regulation of reactive oxygen species biosynthetic process\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0090333\", \"regulation of stomatal closure\"], [\"0009409\", \"response to cold\"], [\"0009269\", \"response to desiccation\"], [\"0010218\", \"response to far red light\"], [\"0009644\", \"response to high light intensity\"], [\"0009416\", \"response to light stimulus\"], [\"0009645\", \"response to low light intensity stimulus\"], [\"0010114\", \"response to red light\"], [\"0030104\", \"water homeostasis\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0051607\", \"defense response to virus\"], [\"0051665\", \"membrane raft localization\"], [\"0009751\", \"response to salicylic acid\"], [\"0009738\", \"abscisic acid-activated signaling pathway\"], [\"0007015\", \"actin filament organization\"], [\"0030865\", \"cortical cytoskeleton organization\"], [\"0007163\", \"establishment or maintenance of cell polarity\"], [\"0009788\", \"negative regulation of abscisic acid-activated signaling pathway\"], [\"0032956\", \"regulation of actin cytoskeleton organization\"], [\"0008360\", \"regulation of cell shape\"], [\"0007264\", \"small GTPase mediated signal transduction\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c240 @network_members=#<Interaction_Network:0x0000561fbd81a8e8 @agi_locus1=\"At1g23740\", @agi_locus2=\"At2g26080\", @interactors=[\"At1g02730\", \"At3g06470\"]>, @kegg=[[\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath00520\", \"Amino sugar and nucleotide sugar metabolism\"]], @go=[[\"0009409\", \"response to cold\"], [\"0046856\", \"phosphatidylinositol dephosphorylation\"], [\"0019464\", \"glycine decarboxylation via glycine cleavage system\"], [\"0006546\", \"glycine catabolic process\"], [\"0000398\", \"mRNA splicing, via spliceosome\"], [\"0000919\", \"cell plate assembly\"], [\"0042546\", \"cell wall biogenesis\"], [\"0071555\", \"cell wall organization\"], [\"0030244\", \"cellulose biosynthetic process\"], [\"0009833\", \"plant-type primary cell wall biogenesis\"], [\"0006970\", \"response to osmotic stress\"], [\"0009651\", \"response to salt stress\"], [\"0009414\", \"response to water deprivation\"], [\"0048367\", \"shoot system development\"], [\"0006782\", \"protoporphyrinogen IX biosynthetic process\"], [\"0042761\", \"very long-chain fatty acid biosynthetic process\"], [\"0006629\", \"lipid metabolic process\"], [\"0034625\", \"fatty acid elongation, monounsaturated fatty acid\"], [\"0034626\", \"fatty acid elongation, polyunsaturated fatty acid\"], [\"0019367\", \"fatty acid elongation, saturated fatty acid\"], [\"0030148\", \"sphingolipid biosynthetic process\"], [\"0006633\", \"fatty acid biosynthetic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c218 @network_members=#<Interaction_Network:0x0000561fbd81a708 @agi_locus1=\"At2g26080\", @agi_locus2=\"At3g54050\", @interactors=[\"At3g62870\"]>, @kegg=[[\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath00010\", \"Glycolysis / Gluconeogenesis\"], [\"ath00030\", \"Pentose phosphate pathway\"], [\"ath00051\", \"Fructose and mannose metabolism\"], [\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath03010\", \"Ribosome\"]], @go=[[\"0019464\", \"glycine decarboxylation via glycine cleavage system\"], [\"0006546\", \"glycine catabolic process\"], [\"0000398\", \"mRNA splicing, via spliceosome\"], [\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006002\", \"fructose 6-phosphate metabolic process\"], [\"0006000\", \"fructose metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0015979\", \"photosynthesis\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0009409\", \"response to cold\"], [\"0005986\", \"sucrose biosynthetic process\"], [\"0005985\", \"sucrose metabolic process\"], [\"0005975\", \"carbohydrate metabolic process\"], [\"0000470\", \"maturation of LSU-rRNA\"], [\"0042254\", \"ribosome biogenesis\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c1f0 @network_members=#<Interaction_Network:0x0000561fbd81a528 @agi_locus1=\"At2g26080\", @agi_locus2=\"At3g45780\", @interactors=[\"At5g38890\"]>, @kegg=[[\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath03018\", \"RNA degradation\"]], @go=[[\"0019464\", \"glycine decarboxylation via glycine cleavage system\"], [\"0006546\", \"glycine catabolic process\"], [\"0000398\", \"mRNA splicing, via spliceosome\"], [\"0009904\", \"chloroplast accumulation movement\"], [\"0009903\", \"chloroplast avoidance movement\"], [\"0007623\", \"circadian rhythm\"], [\"0010362\", \"negative regulation of anion channel activity by blue light\"], [\"0009638\", \"phototropism\"], [\"0046777\", \"protein autophosphorylation\"], [\"0006468\", \"protein phosphorylation\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0010155\", \"regulation of proton transport\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0009637\", \"response to blue light\"], [\"0050896\", \"response to stimulus\"], [\"0006396\", \"RNA processing\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c1c8 @network_members=#<Interaction_Network:0x0000561fbd81a320 @agi_locus1=\"At1g68570\", @agi_locus2=\"At2g26080\", @interactors=[\"At3g62870\"]>, @kegg=[[\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath03010\", \"Ribosome\"]], @go=[[\"0010336\", \"gibberellic acid homeostasis\"], [\"0042128\", \"nitrate assimilation\"], [\"0006857\", \"oligopeptide transport\"], [\"0019464\", \"glycine decarboxylation via glycine cleavage system\"], [\"0006546\", \"glycine catabolic process\"], [\"0000398\", \"mRNA splicing, via spliceosome\"], [\"0000470\", \"maturation of LSU-rRNA\"], [\"0042254\", \"ribosome biogenesis\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c1a0 @network_members=#<Interaction_Network:0x0000561fbd81a0f0 @agi_locus1=\"At3g48360\", @agi_locus2=\"At3g54050\", @interactors=[\"At4g09570\"]>, @kegg=[[\"ath00010\", \"Glycolysis / Gluconeogenesis\"], [\"ath00030\", \"Pentose phosphate pathway\"], [\"ath00051\", \"Fructose and mannose metabolism\"], [\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath04626\", \"Plant-pathogen interaction\"]], @go=[[\"0009738\", \"abscisic acid-activated signaling pathway\"], [\"0009734\", \"auxin-activated signaling pathway\"], [\"0007623\", \"circadian rhythm\"], [\"0051973\", \"positive regulation of telomerase activity\"], [\"0016567\", \"protein ubiquitination\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009737\", \"response to abscisic acid\"], [\"0009733\", \"response to auxin\"], [\"0009743\", \"response to carbohydrate\"], [\"0009409\", \"response to cold\"], [\"0042542\", \"response to hydrogen peroxide\"], [\"0009753\", \"response to jasmonic acid\"], [\"0010167\", \"response to nitrate\"], [\"0009751\", \"response to salicylic acid\"], [\"0009651\", \"response to salt stress\"], [\"0009611\", \"response to wounding\"], [\"0010182\", \"sugar mediated signaling pathway\"], [\"0006857\", \"oligopeptide transport\"], [\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006002\", \"fructose 6-phosphate metabolic process\"], [\"0006000\", \"fructose metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0015979\", \"photosynthesis\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0005986\", \"sucrose biosynthetic process\"], [\"0005985\", \"sucrose metabolic process\"], [\"0005975\", \"carbohydrate metabolic process\"], [\"0035556\", \"intracellular signal transduction\"], [\"0018105\", \"peptidyl-serine phosphorylation\"], [\"0009789\", \"positive regulation of abscisic acid-activated signaling pathway\"], [\"0046777\", \"protein autophosphorylation\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c178 @network_members=#<Interaction_Network:0x0000561fbd819f60 @agi_locus1=\"At3g48360\", @agi_locus2=\"At4g37610\", @interactors=[\"At1g26830\"]>, @kegg=[[\"ath04120\", \"Ubiquitin mediated proteolysis\"]], @go=[[\"0009738\", \"abscisic acid-activated signaling pathway\"], [\"0009734\", \"auxin-activated signaling pathway\"], [\"0007623\", \"circadian rhythm\"], [\"0051973\", \"positive regulation of telomerase activity\"], [\"0016567\", \"protein ubiquitination\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009737\", \"response to abscisic acid\"], [\"0009733\", \"response to auxin\"], [\"0009743\", \"response to carbohydrate\"], [\"0009409\", \"response to cold\"], [\"0042542\", \"response to hydrogen peroxide\"], [\"0009753\", \"response to jasmonic acid\"], [\"0010167\", \"response to nitrate\"], [\"0009751\", \"response to salicylic acid\"], [\"0009651\", \"response to salt stress\"], [\"0009611\", \"response to wounding\"], [\"0010182\", \"sugar mediated signaling pathway\"], [\"0006857\", \"oligopeptide transport\"], [\"0010200\", \"response to chitin\"], [\"0009911\", \"positive regulation of flower development\"], [\"0009639\", \"response to red or far red light\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c150 @network_members=#<Interaction_Network:0x0000561fbd819df8 @agi_locus1=\"At2g43010\", @agi_locus2=\"At3g16250\", @interactors=[\"At2g01570\"]>, @kegg=[[\"ath04075\", \"Plant hormone signal transduction\"]], @go=[[\"0009704\", \"de-etiolation\"], [\"0009638\", \"phototropism\"], [\"0010161\", \"red light signaling pathway\"], [\"0010017\", \"red or far-red light signaling pathway\"], [\"0009585\", \"red, far-red light phototransduction\"], [\"0010600\", \"regulation of auxin biosynthetic process\"], [\"0010928\", \"regulation of auxin mediated signaling pathway\"], [\"0010244\", \"response to low fluence blue light stimulus by blue low-fluence system\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009740\", \"gibberellic acid mediated signaling pathway\"], [\"0042538\", \"hyperosmotic salinity response\"], [\"0009867\", \"jasmonic acid mediated signaling pathway\"], [\"0033206\", \"meiotic cytokinesis\"], [\"0007275\", \"multicellular organism development\"], [\"1905614\", \"negative regulation of developmental vegetative growth\"], [\"0009938\", \"negative regulation of gibberellic acid mediated signaling pathway\"], [\"1905622\", \"negative regulation of leaf development\"], [\"0010187\", \"negative regulation of seed germination\"], [\"1900033\", \"negative regulation of trichome patterning\"], [\"0010628\", \"positive regulation of gene expression\"], [\"0045893\", \"positive regulation of transcription, DNA-templated\"], [\"0042176\", \"regulation of protein catabolic process\"], [\"2000377\", \"regulation of reactive oxygen species metabolic process\"], [\"2000033\", \"regulation of seed dormancy process\"], [\"0010029\", \"regulation of seed germination\"], [\"0009737\", \"response to abscisic acid\"], [\"0009723\", \"response to ethylene\"], [\"0010218\", \"response to far red light\"], [\"0009739\", \"response to gibberellin\"], [\"0009863\", \"salicylic acid mediated signaling pathway\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c128 @network_members=#<Interaction_Network:0x0000561fbd819c90 @agi_locus1=\"At1g69870\", @agi_locus2=\"At3g28220\", @interactors=[\"At1g44960\"]>, @kegg=[], @go=[[\"0042128\", \"nitrate assimilation\"], [\"0015706\", \"nitrate transport\"], [\"0006355\", \"regulation of transcription, DNA-templated\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c100 @network_members=#<Interaction_Network:0x0000561fbd819b28 @agi_locus1=\"At2g47930\", @agi_locus2=\"At3g28220\", @interactors=[\"At3g15710\"]>, @kegg=[[\"ath03060\", \"Protein export\"]], @go=[[\"0080167\", \"response to karrikin\"], [\"0006465\", \"signal peptide processing\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c0d8 @network_members=#<Interaction_Network:0x0000561fbd8199c0 @agi_locus1=\"At3g28220\", @agi_locus2=\"At5g43750\", @interactors=[\"At3g46890\", \"At4g20790\"]>, @kegg=[], @go=[[\"0006979\", \"response to oxidative stress\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c0b0 @network_members=#<Interaction_Network:0x0000561fbd819858 @agi_locus1=\"At3g28220\", @agi_locus2=\"At5g38990\", @interactors=[\"At4g20790\"]>, @kegg=[], @go=[[\"0046777\", \"protein autophosphorylation\"], [\"0010038\", \"response to metal ion\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c088 @network_members=#<Interaction_Network:0x0000561fbd819650 @agi_locus1=\"At2g26740\", @agi_locus2=\"At4g17090\", @interactors=[\"At1g12820\", \"At5g63860\", \"At2g24270\"]>, @kegg=[[\"ath00500\", \"Starch and sucrose metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath00010\", \"Glycolysis / Gluconeogenesis\"], [\"ath00030\", \"Pentose phosphate pathway\"], [\"ath01200\", \"Carbon metabolism\"]], @go=[[\"0009733\", \"response to auxin\"], [\"0009414\", \"response to water deprivation\"], [\"0000024\", \"maltose biosynthetic process\"], [\"0009409\", \"response to cold\"], [\"0005983\", \"starch catabolic process\"], [\"0000272\", \"polysaccharide catabolic process\"], [\"0009734\", \"auxin-activated signaling pathway\"], [\"0071249\", \"cellular response to nitrate\"], [\"0006952\", \"defense response\"], [\"0048527\", \"lateral root development\"], [\"0010152\", \"pollen maturation\"], [\"0080022\", \"primary root development\"], [\"0016567\", \"protein ubiquitination\"], [\"0031146\", \"SCF-dependent proteasomal ubiquitin-dependent protein catabolic process\"], [\"0048443\", \"stamen development\"], [\"0009649\", \"entrainment of circadian clock\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009411\", \"response to UV\"], [\"0010224\", \"response to UV-B\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c060 @network_members=#<Interaction_Network:0x0000561fbd819420 @agi_locus1=\"At2g26740\", @agi_locus2=\"At5g48300\", @interactors=[\"At3g17240\"]>, @kegg=[[\"ath00500\", \"Starch and sucrose metabolism\"], [\"ath00520\", \"Amino sugar and nucleotide sugar metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01250\", \"Biosynthesis of nucleotide sugars\"], [\"ath00010\", \"Glycolysis / Gluconeogenesis\"], [\"ath00020\", \"Citrate cycle (TCA cycle)\"], [\"ath00260\", \"Glycine, serine and threonine metabolism\"], [\"ath00280\", \"Valine, leucine and isoleucine degradation\"], [\"ath00310\", \"Lysine degradation\"], [\"ath00380\", \"Tryptophan metabolism\"], [\"ath00620\", \"Pyruvate metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00640\", \"Propanoate metabolism\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath01240\", \"Biosynthesis of cofactors\"]], @go=[[\"0009733\", \"response to auxin\"], [\"0009414\", \"response to water deprivation\"], [\"0005978\", \"glycogen biosynthetic process\"], [\"0048573\", \"photoperiodism, flowering\"], [\"0019252\", \"starch biosynthetic process\"], [\"0006457\", \"protein folding\"], [\"0045454\", \"cell redox homeostasis\"], [\"0046686\", \"response to cadmium ion\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c038 @network_members=#<Interaction_Network:0x0000561fbd819218 @agi_locus1=\"At1g68570\", @agi_locus2=\"At3g54050\", @interactors=[\"At3g62870\"]>, @kegg=[[\"ath00010\", \"Glycolysis / Gluconeogenesis\"], [\"ath00030\", \"Pentose phosphate pathway\"], [\"ath00051\", \"Fructose and mannose metabolism\"], [\"ath00710\", \"Carbon fixation in photosynthetic organisms\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01110\", \"Biosynthesis of secondary metabolites\"], [\"ath01200\", \"Carbon metabolism\"], [\"ath03010\", \"Ribosome\"]], @go=[[\"0010336\", \"gibberellic acid homeostasis\"], [\"0042128\", \"nitrate assimilation\"], [\"0006857\", \"oligopeptide transport\"], [\"0030388\", \"fructose 1,6-bisphosphate metabolic process\"], [\"0006002\", \"fructose 6-phosphate metabolic process\"], [\"0006000\", \"fructose metabolic process\"], [\"0006094\", \"gluconeogenesis\"], [\"0015979\", \"photosynthesis\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0009409\", \"response to cold\"], [\"0005986\", \"sucrose biosynthetic process\"], [\"0005985\", \"sucrose metabolic process\"], [\"0005975\", \"carbohydrate metabolic process\"], [\"0000470\", \"maturation of LSU-rRNA\"], [\"0042254\", \"ribosome biogenesis\"]]>, #<Uso_General_Annotation:0x0000561fbcc0c010 @network_members=#<Interaction_Network:0x0000561fbd818e08 @agi_locus1=\"At1g14280\", @agi_locus2=\"At3g45780\", @interactors=[\"At2g02950\"]>, @kegg=[], @go=[[\"0048366\", \"leaf development\"], [\"0009638\", \"phototropism\"], [\"0009585\", \"red, far-red light phototransduction\"], [\"0009904\", \"chloroplast accumulation movement\"], [\"0009903\", \"chloroplast avoidance movement\"], [\"0007623\", \"circadian rhythm\"], [\"0010362\", \"negative regulation of anion channel activity by blue light\"], [\"0046777\", \"protein autophosphorylation\"], [\"0006468\", \"protein phosphorylation\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0010155\", \"regulation of proton transport\"], [\"0010119\", \"regulation of stomatal movement\"], [\"0009637\", \"response to blue light\"], [\"0050896\", \"response to stimulus\"], [\"0009958\", \"positive gravitropism\"], [\"0010017\", \"red or far-red light signaling pathway\"], [\"0010218\", \"response to far red light\"], [\"0010114\", \"response to red light\"]]>, #<Uso_General_Annotation:0x0000561fbc0dff98 @network_members=#<Interaction_Network:0x0000561fbd818bd8 @agi_locus1=\"At2g30950\", @agi_locus2=\"At3g62410\", @interactors=[\"At1g22920\"]>, @kegg=[], @go=[[\"0009658\", \"chloroplast organization\"], [\"0007275\", \"multicellular organism development\"], [\"0010205\", \"photoinhibition\"], [\"0048564\", \"photosystem I assembly\"], [\"0010206\", \"photosystem II repair\"], [\"0030163\", \"protein catabolic process\"], [\"0006508\", \"proteolysis\"], [\"0010304\", \"PSII associated light-harvesting complex II catabolic process\"], [\"0072593\", \"reactive oxygen species metabolic process\"], [\"0010027\", \"thylakoid membrane organization\"], [\"0071454\", \"cellular response to anoxia\"], [\"0070417\", \"cellular response to cold\"], [\"0034605\", \"cellular response to heat\"], [\"0080153\", \"negative regulation of reductive pentose-phosphate cycle\"], [\"0018316\", \"peptide cross-linking via L-cystine\"], [\"0065003\", \"protein-containing complex assembly\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0010387\", \"COP9 signalosome assembly\"], [\"0010100\", \"negative regulation of photomorphogenesis\"], [\"0010971\", \"positive regulation of G2/M transition of mitotic cell cycle\"], [\"0000338\", \"protein deneddylation\"], [\"0009585\", \"red, far-red light phototransduction\"], [\"0031347\", \"regulation of defense response\"], [\"0009733\", \"response to auxin\"], [\"0010093\", \"specification of floral organ identity\"]]>, #<Uso_General_Annotation:0x0000561fbc0dff70 @network_members=#<Interaction_Network:0x0000561fbd818a70 @agi_locus1=\"At2g30950\", @agi_locus2=\"At3g16250\", @interactors=[\"At1g22920\"]>, @kegg=[], @go=[[\"0009658\", \"chloroplast organization\"], [\"0007275\", \"multicellular organism development\"], [\"0010205\", \"photoinhibition\"], [\"0048564\", \"photosystem I assembly\"], [\"0010206\", \"photosystem II repair\"], [\"0030163\", \"protein catabolic process\"], [\"0006508\", \"proteolysis\"], [\"0010304\", \"PSII associated light-harvesting complex II catabolic process\"], [\"0072593\", \"reactive oxygen species metabolic process\"], [\"0010027\", \"thylakoid membrane organization\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0010387\", \"COP9 signalosome assembly\"], [\"0010100\", \"negative regulation of photomorphogenesis\"], [\"0010971\", \"positive regulation of G2/M transition of mitotic cell cycle\"], [\"0000338\", \"protein deneddylation\"], [\"0009585\", \"red, far-red light phototransduction\"], [\"0031347\", \"regulation of defense response\"], [\"0009733\", \"response to auxin\"], [\"0010093\", \"specification of floral organ identity\"]]>, #<Uso_General_Annotation:0x0000561fbc0dff48 @network_members=#<Interaction_Network:0x0000561fbd818908 @agi_locus1=\"At3g16250\", @agi_locus2=\"At3g62410\", @interactors=[\"At1g22920\"]>, @kegg=[], @go=[[\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0071454\", \"cellular response to anoxia\"], [\"0070417\", \"cellular response to cold\"], [\"0034605\", \"cellular response to heat\"], [\"0080153\", \"negative regulation of reductive pentose-phosphate cycle\"], [\"0018316\", \"peptide cross-linking via L-cystine\"], [\"0065003\", \"protein-containing complex assembly\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0010387\", \"COP9 signalosome assembly\"], [\"0010100\", \"negative regulation of photomorphogenesis\"], [\"0010971\", \"positive regulation of G2/M transition of mitotic cell cycle\"], [\"0000338\", \"protein deneddylation\"], [\"0009585\", \"red, far-red light phototransduction\"], [\"0031347\", \"regulation of defense response\"], [\"0009733\", \"response to auxin\"], [\"0010093\", \"specification of floral organ identity\"]]>, #<Uso_General_Annotation:0x0000561fbc0dff20 @network_members=#<Interaction_Network:0x0000561fbd8187a0 @agi_locus1=\"At3g62410\", @agi_locus2=\"At5g35630\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00220\", \"Arginine biosynthesis\"], [\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath01230\", \"Biosynthesis of amino acids\"]], @go=[[\"0071454\", \"cellular response to anoxia\"], [\"0070417\", \"cellular response to cold\"], [\"0034605\", \"cellular response to heat\"], [\"0080153\", \"negative regulation of reductive pentose-phosphate cycle\"], [\"0018316\", \"peptide cross-linking via L-cystine\"], [\"0065003\", \"protein-containing complex assembly\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0007568\", \"aging\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006542\", \"glutamine biosynthetic process\"], [\"0046686\", \"response to cadmium ion\"], [\"0006807\", \"nitrogen compound metabolic process\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfef8 @network_members=#<Interaction_Network:0x0000561fbd818638 @agi_locus1=\"At3g50820\", @agi_locus2=\"At3g62410\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0010205\", \"photoinhibition\"], [\"0019684\", \"photosynthesis, light reaction\"], [\"0010207\", \"photosystem II assembly\"], [\"0042549\", \"photosystem II stabilization\"], [\"0035304\", \"regulation of protein dephosphorylation\"], [\"0071454\", \"cellular response to anoxia\"], [\"0070417\", \"cellular response to cold\"], [\"0034605\", \"cellular response to heat\"], [\"0080153\", \"negative regulation of reductive pentose-phosphate cycle\"], [\"0018316\", \"peptide cross-linking via L-cystine\"], [\"0065003\", \"protein-containing complex assembly\"], [\"0019253\", \"reductive pentose-phosphate cycle\"], [\"0009416\", \"response to light stimulus\"], [\"0009744\", \"response to sucrose\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfed0 @network_members=#<Interaction_Network:0x0000561fbd818480 @agi_locus1=\"At1g03130\", @agi_locus2=\"At3g16250\", @interactors=[\"At3g48150\", \"At3g47620\", \"At3g02150\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"]], @go=[[\"0015979\", \"photosynthesis\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0031145\", \"anaphase-promoting complex-dependent catabolic process\"], [\"0051301\", \"cell division\"], [\"0007091\", \"metaphase/anaphase transition of mitotic cell cycle\"], [\"0045842\", \"positive regulation of mitotic metaphase/anaphase transition\"], [\"0016567\", \"protein ubiquitination\"], [\"0031347\", \"regulation of defense response\"], [\"0007049\", \"cell cycle\"], [\"0030071\", \"regulation of mitotic metaphase/anaphase transition\"], [\"0010229\", \"inflorescence development\"], [\"0010029\", \"regulation of seed germination\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"], [\"0048366\", \"leaf development\"], [\"0010150\", \"leaf senescence\"], [\"2000032\", \"regulation of secondary shoot formation\"], [\"0006021\", \"inositol biosynthetic process\"], [\"0046855\", \"inositol phosphate dephosphorylation\"], [\"0046854\", \"phosphatidylinositol phosphorylation\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfea8 @network_members=#<Interaction_Network:0x0000561fbd8181b0 @agi_locus1=\"At1g29910\", @agi_locus2=\"At3g16250\", @interactors=[\"At3g47620\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009750\", \"response to fructose\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0009733\", \"response to auxin\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0010229\", \"inflorescence development\"], [\"0031347\", \"regulation of defense response\"], [\"0010029\", \"regulation of seed germination\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfe80 @network_members=#<Interaction_Network:0x0000561fbd818e30 @agi_locus1=\"At1g29920\", @agi_locus2=\"At3g16250\", @interactors=[\"At3g47620\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009750\", \"response to fructose\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0010229\", \"inflorescence development\"], [\"0031347\", \"regulation of defense response\"], [\"0010029\", \"regulation of seed germination\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfe58 @network_members=#<Interaction_Network:0x0000561fbd819448 @agi_locus1=\"At3g16250\", @agi_locus2=\"At4g02770\", @interactors=[\"At3g48150\", \"At3g47620\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"]], @go=[[\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0140547\", \"acquisition of seed longevity\"], [\"0015979\", \"photosynthesis\"], [\"0035556\", \"intracellular signal transduction\"], [\"0031145\", \"anaphase-promoting complex-dependent catabolic process\"], [\"0051301\", \"cell division\"], [\"0007091\", \"metaphase/anaphase transition of mitotic cell cycle\"], [\"0045842\", \"positive regulation of mitotic metaphase/anaphase transition\"], [\"0016567\", \"protein ubiquitination\"], [\"0031347\", \"regulation of defense response\"], [\"0007049\", \"cell cycle\"], [\"0030071\", \"regulation of mitotic metaphase/anaphase transition\"], [\"0010229\", \"inflorescence development\"], [\"0010029\", \"regulation of seed germination\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfe30 @network_members=#<Interaction_Network:0x0000561fbd81a0a0 @agi_locus1=\"At3g16250\", @agi_locus2=\"At4g09010\", @interactors=[\"At3g53900\"]>, @kegg=[[\"ath00053\", \"Ascorbate and aldarate metabolism\"], [\"ath00480\", \"Glutathione metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00240\", \"Pyrimidine metabolism\"]], @go=[[\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0034599\", \"cellular response to oxidative stress\"], [\"0042744\", \"hydrogen peroxide catabolic process\"], [\"0000302\", \"response to reactive oxygen species\"], [\"0006979\", \"response to oxidative stress\"], [\"0016036\", \"cellular response to phosphate starvation\"], [\"0032502\", \"developmental process\"], [\"0009116\", \"nucleoside metabolic process\"], [\"0044206\", \"UMP salvage\"], [\"0006223\", \"uracil salvage\"], [\"0019252\", \"starch biosynthetic process\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfe08 @network_members=#<Interaction_Network:0x0000561fbd81a4d8 @agi_locus1=\"At1g09340\", @agi_locus2=\"At3g63140\", @interactors=\"direct\">, @kegg=[], @go=[[\"0042631\", \"cellular response to water deprivation\"], [\"0009658\", \"chloroplast organization\"], [\"0007623\", \"circadian rhythm\"], [\"0042742\", \"defense response to bacterium\"], [\"0005996\", \"monosaccharide metabolic process\"], [\"0032544\", \"plastid translation\"], [\"0000272\", \"polysaccharide catabolic process\"], [\"0045893\", \"positive regulation of transcription, DNA-templated\"], [\"0045727\", \"positive regulation of translation\"], [\"0010468\", \"regulation of gene expression\"], [\"0009409\", \"response to cold\"], [\"0009611\", \"response to wounding\"], [\"0006364\", \"rRNA processing\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfde0 @network_members=#<Interaction_Network:0x0000561fbd81aaf0 @agi_locus1=\"At3g50820\", @agi_locus2=\"At5g35630\", @interactors=[\"At5g42980\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00220\", \"Arginine biosynthesis\"], [\"ath00250\", \"Alanine, aspartate and glutamate metabolism\"], [\"ath00630\", \"Glyoxylate and dicarboxylate metabolism\"], [\"ath00910\", \"Nitrogen metabolism\"], [\"ath01230\", \"Biosynthesis of amino acids\"]], @go=[[\"0010205\", \"photoinhibition\"], [\"0019684\", \"photosynthesis, light reaction\"], [\"0010207\", \"photosystem II assembly\"], [\"0042549\", \"photosystem II stabilization\"], [\"0035304\", \"regulation of protein dephosphorylation\"], [\"0007568\", \"aging\"], [\"0019676\", \"ammonia assimilation cycle\"], [\"0006542\", \"glutamine biosynthetic process\"], [\"0046686\", \"response to cadmium ion\"], [\"0006807\", \"nitrogen compound metabolic process\"], [\"0050832\", \"defense response to fungus\"], [\"0010286\", \"heat acclimation\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0010188\", \"response to microbial phytotoxin\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfdb8 @network_members=#<Interaction_Network:0x0000561fbd81b130 @agi_locus1=\"At1g29910\", @agi_locus2=\"At1g29920\", @interactors=[\"At3g47620\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009750\", \"response to fructose\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0009733\", \"response to auxin\"], [\"0010229\", \"inflorescence development\"], [\"0031347\", \"regulation of defense response\"], [\"0010029\", \"regulation of seed germination\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfd90 @network_members=#<Interaction_Network:0x0000561fbd81b798 @agi_locus1=\"At1g29910\", @agi_locus2=\"At4g02770\", @interactors=[\"At3g47620\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00195\", \"Photosynthesis\"]], @go=[[\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009750\", \"response to fructose\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0009733\", \"response to auxin\"], [\"0140547\", \"acquisition of seed longevity\"], [\"0015979\", \"photosynthesis\"], [\"0035556\", \"intracellular signal transduction\"], [\"0010229\", \"inflorescence development\"], [\"0031347\", \"regulation of defense response\"], [\"0010029\", \"regulation of seed germination\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfd68 @network_members=#<Interaction_Network:0x0000561fbd81bcc0 @agi_locus1=\"At1g69870\", @agi_locus2=\"At4g22890\", @interactors=[\"At1g17280\", \"At2g22425\", \"At5g37050\", \"At4g37680\", \"At3g12180\", \"At1g29060\"]>, @kegg=[[\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"], [\"ath03060\", \"Protein export\"], [\"ath04130\", \"SNARE interactions in vesicular transport\"]], @go=[[\"0042128\", \"nitrate assimilation\"], [\"0015706\", \"nitrate transport\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0042631\", \"cellular response to water deprivation\"], [\"1902457\", \"negative regulation of stomatal opening\"], [\"0000209\", \"protein polyubiquitination\"], [\"0030433\", \"ubiquitin-dependent ERAD pathway\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"], [\"0045047\", \"protein targeting to ER\"], [\"0006465\", \"signal peptide processing\"], [\"0048658\", \"anther wall tapetum development\"], [\"0009725\", \"response to hormone\"], [\"0009744\", \"response to sucrose\"], [\"0009738\", \"abscisic acid-activated signaling pathway\"], [\"0016192\", \"vesicle-mediated transport\"], [\"1901002\", \"positive regulation of response to salt stress\"], [\"0015031\", \"protein transport\"], [\"0047484\", \"regulation of response to osmotic stress\"], [\"0043182\", \"vacuolar sequestering of sodium ion\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfd40 @network_members=#<Interaction_Network:0x0000561fbd818098 @agi_locus1=\"At1g69870\", @agi_locus2=\"At5g43750\", @interactors=[\"At1g17280\", \"At2g22425\", \"At5g37050\", \"At3g12180\", \"At1g29060\"]>, @kegg=[[\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"], [\"ath03060\", \"Protein export\"], [\"ath04130\", \"SNARE interactions in vesicular transport\"]], @go=[[\"0042128\", \"nitrate assimilation\"], [\"0015706\", \"nitrate transport\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0006979\", \"response to oxidative stress\"], [\"0042631\", \"cellular response to water deprivation\"], [\"1902457\", \"negative regulation of stomatal opening\"], [\"0000209\", \"protein polyubiquitination\"], [\"0030433\", \"ubiquitin-dependent ERAD pathway\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"], [\"0045047\", \"protein targeting to ER\"], [\"0006465\", \"signal peptide processing\"], [\"0048658\", \"anther wall tapetum development\"], [\"0016192\", \"vesicle-mediated transport\"], [\"1901002\", \"positive regulation of response to salt stress\"], [\"0015031\", \"protein transport\"], [\"0047484\", \"regulation of response to osmotic stress\"], [\"0043182\", \"vacuolar sequestering of sodium ion\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfd18 @network_members=#<Interaction_Network:0x0000561fbd818250 @agi_locus1=\"At1g69870\", @agi_locus2=\"At3g61260\", @interactors=[\"At5g37050\", \"At3g12180\", \"At2g26180\"]>, @kegg=[], @go=[[\"0042128\", \"nitrate assimilation\"], [\"0015706\", \"nitrate transport\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0051607\", \"defense response to virus\"], [\"0051665\", \"membrane raft localization\"], [\"0009751\", \"response to salicylic acid\"], [\"0016192\", \"vesicle-mediated transport\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfcf0 @network_members=#<Interaction_Network:0x0000561fbd818368 @agi_locus1=\"At1g69870\", @agi_locus2=\"At5g38990\", @interactors=[\"At1g17280\", \"At4g14455\", \"At1g14700\", \"At5g06320\", \"At1g47640\", \"At1g07860\", \"At5g11890\", \"At4g30850\", \"At3g17000\", \"At5g37050\", \"At3g12180\", \"At2g26180\", \"At1g29060\"]>, @kegg=[[\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"], [\"ath04130\", \"SNARE interactions in vesicular transport\"], [\"ath00740\", \"Riboflavin metabolism\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0042128\", \"nitrate assimilation\"], [\"0015706\", \"nitrate transport\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0046777\", \"protein autophosphorylation\"], [\"0010038\", \"response to metal ion\"], [\"0042631\", \"cellular response to water deprivation\"], [\"1902457\", \"negative regulation of stomatal opening\"], [\"0000209\", \"protein polyubiquitination\"], [\"0030433\", \"ubiquitin-dependent ERAD pathway\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"], [\"0006888\", \"endoplasmic reticulum to Golgi vesicle-mediated transport\"], [\"0015031\", \"protein transport\"], [\"0071456\", \"cellular response to hypoxia\"], [\"0042742\", \"defense response to bacterium\"], [\"0051607\", \"defense response to virus\"], [\"0009617\", \"response to bacterium\"], [\"0009751\", \"response to salicylic acid\"], [\"0009611\", \"response to wounding\"], [\"0009725\", \"response to hormone\"], [\"0009744\", \"response to sucrose\"], [\"0006265\", \"DNA topological change\"], [\"0016192\", \"vesicle-mediated transport\"], [\"1901002\", \"positive regulation of response to salt stress\"], [\"0047484\", \"regulation of response to osmotic stress\"], [\"0043182\", \"vacuolar sequestering of sodium ion\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfcc8 @network_members=#<Interaction_Network:0x0000561fbd8184a8 @agi_locus1=\"At4g22890\", @agi_locus2=\"At5g43750\", @interactors=[\"At5g59650\", \"At1g21240\", \"At5g37050\", \"At1g29060\", \"At1g17280\", \"At3g12180\", \"At4g20790\", \"At2g22425\"]>, @kegg=[[\"ath04130\", \"SNARE interactions in vesicular transport\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"], [\"ath03060\", \"Protein export\"]], @go=[[\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0006979\", \"response to oxidative stress\"], [\"0007166\", \"cell surface receptor signaling pathway\"], [\"0006468\", \"protein phosphorylation\"], [\"1901002\", \"positive regulation of response to salt stress\"], [\"0015031\", \"protein transport\"], [\"0047484\", \"regulation of response to osmotic stress\"], [\"0043182\", \"vacuolar sequestering of sodium ion\"], [\"0016192\", \"vesicle-mediated transport\"], [\"0042631\", \"cellular response to water deprivation\"], [\"1902457\", \"negative regulation of stomatal opening\"], [\"0000209\", \"protein polyubiquitination\"], [\"0030433\", \"ubiquitin-dependent ERAD pathway\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"], [\"0045047\", \"protein targeting to ER\"], [\"0006465\", \"signal peptide processing\"], [\"0048658\", \"anther wall tapetum development\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfca0 @network_members=#<Interaction_Network:0x0000561fbd818570 @agi_locus1=\"At1g69870\", @agi_locus2=\"At5g43750\", @interactors=[\"At5g37050\", \"At1g29060\", \"At1g17280\", \"At3g12180\", \"At2g22425\"]>, @kegg=[[\"ath04130\", \"SNARE interactions in vesicular transport\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"], [\"ath03060\", \"Protein export\"]], @go=[[\"0042128\", \"nitrate assimilation\"], [\"0015706\", \"nitrate transport\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0006979\", \"response to oxidative stress\"], [\"1901002\", \"positive regulation of response to salt stress\"], [\"0015031\", \"protein transport\"], [\"0047484\", \"regulation of response to osmotic stress\"], [\"0043182\", \"vacuolar sequestering of sodium ion\"], [\"0016192\", \"vesicle-mediated transport\"], [\"0042631\", \"cellular response to water deprivation\"], [\"1902457\", \"negative regulation of stomatal opening\"], [\"0000209\", \"protein polyubiquitination\"], [\"0030433\", \"ubiquitin-dependent ERAD pathway\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"], [\"0045047\", \"protein targeting to ER\"], [\"0006465\", \"signal peptide processing\"], [\"0048658\", \"anther wall tapetum development\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfc78 @network_members=#<Interaction_Network:0x0000561fbd818660 @agi_locus1=\"At3g61260\", @agi_locus2=\"At5g43750\", @interactors=[\"At1g21240\", \"At5g37050\", \"At3g12180\"]>, @kegg=[], @go=[[\"0051607\", \"defense response to virus\"], [\"0051665\", \"membrane raft localization\"], [\"0009751\", \"response to salicylic acid\"], [\"0006979\", \"response to oxidative stress\"], [\"0007166\", \"cell surface receptor signaling pathway\"], [\"0006468\", \"protein phosphorylation\"], [\"0016192\", \"vesicle-mediated transport\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfc50 @network_members=#<Interaction_Network:0x0000561fbd818700 @agi_locus1=\"At5g38990\", @agi_locus2=\"At5g43750\", @interactors=[\"At5g59650\", \"At5g37050\", \"At1g29060\", \"At1g17280\", \"At3g12180\", \"At4g20790\"]>, @kegg=[[\"ath04130\", \"SNARE interactions in vesicular transport\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"]], @go=[[\"0046777\", \"protein autophosphorylation\"], [\"0010038\", \"response to metal ion\"], [\"0006979\", \"response to oxidative stress\"], [\"1901002\", \"positive regulation of response to salt stress\"], [\"0015031\", \"protein transport\"], [\"0047484\", \"regulation of response to osmotic stress\"], [\"0043182\", \"vacuolar sequestering of sodium ion\"], [\"0016192\", \"vesicle-mediated transport\"], [\"0042631\", \"cellular response to water deprivation\"], [\"1902457\", \"negative regulation of stomatal opening\"], [\"0000209\", \"protein polyubiquitination\"], [\"0030433\", \"ubiquitin-dependent ERAD pathway\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfc28 @network_members=#<Interaction_Network:0x0000561fbd8187c8 @agi_locus1=\"At1g69870\", @agi_locus2=\"At3g61260\", @interactors=[\"At2g26180\", \"At5g37050\", \"At3g12180\"]>, @kegg=[], @go=[[\"0042128\", \"nitrate assimilation\"], [\"0015706\", \"nitrate transport\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0051607\", \"defense response to virus\"], [\"0051665\", \"membrane raft localization\"], [\"0009751\", \"response to salicylic acid\"], [\"0016192\", \"vesicle-mediated transport\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfc00 @network_members=#<Interaction_Network:0x0000561fbd818868 @agi_locus1=\"At3g61260\", @agi_locus2=\"At5g38990\", @interactors=[\"At2g26180\", \"At5g37050\", \"At3g12180\"]>, @kegg=[], @go=[[\"0051607\", \"defense response to virus\"], [\"0051665\", \"membrane raft localization\"], [\"0009751\", \"response to salicylic acid\"], [\"0046777\", \"protein autophosphorylation\"], [\"0010038\", \"response to metal ion\"], [\"0016192\", \"vesicle-mediated transport\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfbd8 @network_members=#<Interaction_Network:0x0000561fbd818930 @agi_locus1=\"At4g22890\", @agi_locus2=\"At5g38990\", @interactors=[\"At1g29060\", \"At5g37050\", \"At1g17280\", \"At3g12180\", \"At5g59650\", \"At4g20790\"]>, @kegg=[[\"ath04130\", \"SNARE interactions in vesicular transport\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"]], @go=[[\"0009773\", \"photosynthetic electron transport in photosystem I\"], [\"0046777\", \"protein autophosphorylation\"], [\"0010038\", \"response to metal ion\"], [\"1901002\", \"positive regulation of response to salt stress\"], [\"0015031\", \"protein transport\"], [\"0047484\", \"regulation of response to osmotic stress\"], [\"0043182\", \"vacuolar sequestering of sodium ion\"], [\"0016192\", \"vesicle-mediated transport\"], [\"0042631\", \"cellular response to water deprivation\"], [\"1902457\", \"negative regulation of stomatal opening\"], [\"0000209\", \"protein polyubiquitination\"], [\"0030433\", \"ubiquitin-dependent ERAD pathway\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfbb0 @network_members=#<Interaction_Network:0x0000561fbd8189d0 @agi_locus1=\"At1g69870\", @agi_locus2=\"At5g38990\", @interactors=[\"At2g26180\", \"At1g14700\", \"At5g11890\", \"At1g29060\", \"At1g07860\", \"At5g37050\", \"At1g17280\", \"At3g12180\", \"At3g17000\", \"At5g06320\", \"At4g14455\", \"At4g30850\", \"At1g47640\"]>, @kegg=[[\"ath00740\", \"Riboflavin metabolism\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath04130\", \"SNARE interactions in vesicular transport\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"]], @go=[[\"0042128\", \"nitrate assimilation\"], [\"0015706\", \"nitrate transport\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0046777\", \"protein autophosphorylation\"], [\"0010038\", \"response to metal ion\"], [\"1901002\", \"positive regulation of response to salt stress\"], [\"0015031\", \"protein transport\"], [\"0047484\", \"regulation of response to osmotic stress\"], [\"0043182\", \"vacuolar sequestering of sodium ion\"], [\"0016192\", \"vesicle-mediated transport\"], [\"0042631\", \"cellular response to water deprivation\"], [\"1902457\", \"negative regulation of stomatal opening\"], [\"0000209\", \"protein polyubiquitination\"], [\"0030433\", \"ubiquitin-dependent ERAD pathway\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"], [\"0071456\", \"cellular response to hypoxia\"], [\"0042742\", \"defense response to bacterium\"], [\"0051607\", \"defense response to virus\"], [\"0009617\", \"response to bacterium\"], [\"0009751\", \"response to salicylic acid\"], [\"0009611\", \"response to wounding\"], [\"0006888\", \"endoplasmic reticulum to Golgi vesicle-mediated transport\"], [\"0009725\", \"response to hormone\"], [\"0009744\", \"response to sucrose\"], [\"0006265\", \"DNA topological change\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfb88 @network_members=#<Interaction_Network:0x0000561fbd818a98 @agi_locus1=\"At5g38990\", @agi_locus2=\"At5g43750\", @interactors=[\"At1g29060\", \"At5g37050\", \"At1g17280\", \"At3g12180\", \"At5g59650\", \"At4g20790\"]>, @kegg=[[\"ath04130\", \"SNARE interactions in vesicular transport\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"]], @go=[[\"0046777\", \"protein autophosphorylation\"], [\"0010038\", \"response to metal ion\"], [\"0006979\", \"response to oxidative stress\"], [\"1901002\", \"positive regulation of response to salt stress\"], [\"0015031\", \"protein transport\"], [\"0047484\", \"regulation of response to osmotic stress\"], [\"0043182\", \"vacuolar sequestering of sodium ion\"], [\"0016192\", \"vesicle-mediated transport\"], [\"0042631\", \"cellular response to water deprivation\"], [\"1902457\", \"negative regulation of stomatal opening\"], [\"0000209\", \"protein polyubiquitination\"], [\"0030433\", \"ubiquitin-dependent ERAD pathway\"], [\"0006511\", \"ubiquitin-dependent protein catabolic process\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfb60 @network_members=#<Interaction_Network:0x0000561fbd818b38 @agi_locus1=\"At1g29920\", @agi_locus2=\"At4g02770\", @interactors=[\"At3g47620\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath00195\", \"Photosynthesis\"]], @go=[[\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009750\", \"response to fructose\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0140547\", \"acquisition of seed longevity\"], [\"0015979\", \"photosynthesis\"], [\"0035556\", \"intracellular signal transduction\"], [\"0010229\", \"inflorescence development\"], [\"0031347\", \"regulation of defense response\"], [\"0010029\", \"regulation of seed germination\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"], [\"0009739\", \"response to gibberellin\"], [\"0015031\", \"protein transport\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfb38 @network_members=#<Interaction_Network:0x0000561fbd818c00 @agi_locus1=\"At1g29920\", @agi_locus2=\"At5g01530\", @interactors=[\"At3g26090\"]>, @kegg=[[\"ath00196\", \"Photosynthesis - antenna proteins\"], [\"ath01100\", \"Metabolic pathways\"]], @go=[[\"0009768\", \"photosynthesis, light harvesting in photosystem I\"], [\"0018298\", \"protein-chromophore linkage\"], [\"0009416\", \"response to light stimulus\"], [\"0009750\", \"response to fructose\"], [\"0009765\", \"photosynthesis, light harvesting\"], [\"0009867\", \"jasmonic acid mediated signaling pathway\"], [\"0009968\", \"negative regulation of signal transduction\"], [\"0043547\", \"positive regulation of GTPase activity\"], [\"0042127\", \"regulation of cell population proliferation\"], [\"0009737\", \"response to abscisic acid\"], [\"0009749\", \"response to glucose\"], [\"0009414\", \"response to water deprivation\"], [\"0010182\", \"sugar mediated signaling pathway\"], [\"0006355\", \"regulation of transcription, DNA-templated\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfb10 @network_members=#<Interaction_Network:0x0000561fbd818cc8 @agi_locus1=\"At1g03130\", @agi_locus2=\"At4g02770\", @interactors=[\"At3g08530\", \"At3g48150\", \"At5g37670\", \"At4g19700\", \"At5g32440\", \"At4g29930\", \"At5g61010\", \"At3g47620\", \"At2g23290\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath04144\", \"Endocytosis\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"], [\"ath04141\", \"Protein processing in endoplasmic reticulum\"]], @go=[[\"0015979\", \"photosynthesis\"], [\"0140547\", \"acquisition of seed longevity\"], [\"0035556\", \"intracellular signal transduction\"], [\"0006897\", \"endocytosis\"], [\"0006886\", \"intracellular protein transport\"], [\"0006898\", \"receptor-mediated endocytosis\"], [\"0016192\", \"vesicle-mediated transport\"], [\"0006606\", \"protein import into nucleus\"], [\"0031145\", \"anaphase-promoting complex-dependent catabolic process\"], [\"0051301\", \"cell division\"], [\"0007091\", \"metaphase/anaphase transition of mitotic cell cycle\"], [\"0045842\", \"positive regulation of mitotic metaphase/anaphase transition\"], [\"0016567\", \"protein ubiquitination\"], [\"0031347\", \"regulation of defense response\"], [\"0007049\", \"cell cycle\"], [\"0030071\", \"regulation of mitotic metaphase/anaphase transition\"], [\"0071456\", \"cellular response to hypoxia\"], [\"0051259\", \"protein complex oligomerization\"], [\"0006457\", \"protein folding\"], [\"0009408\", \"response to heat\"], [\"0042542\", \"response to hydrogen peroxide\"], [\"0000302\", \"response to reactive oxygen species\"], [\"0009651\", \"response to salt stress\"], [\"0042742\", \"defense response to bacterium\"], [\"0050832\", \"defense response to fungus\"], [\"2000117\", \"negative regulation of cysteine-type endopeptidase activity\"], [\"0043069\", \"negative regulation of programmed cell death\"], [\"0043161\", \"proteasome-mediated ubiquitin-dependent protein catabolic process\"], [\"0043067\", \"regulation of programmed cell death\"], [\"0009739\", \"response to gibberellin\"], [\"0006355\", \"regulation of transcription, DNA-templated\"], [\"0048268\", \"clathrin coat assembly\"], [\"0052542\", \"defense response by callose deposition\"], [\"0006887\", \"exocytosis\"], [\"0006955\", \"immune response\"], [\"1903553\", \"positive regulation of extracellular exosome assembly\"], [\"0015031\", \"protein transport\"], [\"1903533\", \"regulation of protein targeting\"], [\"0010229\", \"inflorescence development\"], [\"0010029\", \"regulation of seed germination\"], [\"0009737\", \"response to abscisic acid\"], [\"0009735\", \"response to cytokinin\"]]>, #<Uso_General_Annotation:0x0000561fbc0dfae8 @network_members=#<Interaction_Network:0x0000561fbd818d68 @agi_locus1=\"At2g17880\", @agi_locus2=\"At4g02770\", @interactors=[\"At3g49580\", \"At3g48150\", \"At4g17680\"]>, @kegg=[[\"ath00195\", \"Photosynthesis\"], [\"ath01100\", \"Metabolic pathways\"], [\"ath04120\", \"Ubiquitin mediated proteolysis\"]], @go=[[\"0140547\", \"acquisition of seed longevity\"], [\"0015979\", \"photosynthesis\"], [\"0035556\", \"intracellular signal transduction\"], [\"0098869\", \"cellular oxidant detoxification\"], [\"0010438\", \"cellular response to sulfur starvation\"], [\"0009651\", \"response to salt stress\"], [\"0008654\", \"phospholipid biosynthetic process\"], [\"0031145\", \"anaphase-promoting complex-dependent catabolic process\"], [\"0051301\", \"cell division\"], [\"0007091\", \"metaphase/anaphase transition of mitotic cell cycle\"], [\"0045842\", \"positive regulation of mitotic metaphase/anaphase transition\"], [\"0016567\", \"protein ubiquitination\"], [\"0031347\", \"regulation of defense response\"], [\"0007049\", \"cell cycle\"], [\"0030071\", \"regulation of mitotic metaphase/anaphase transition\"], [\"0006952\", \"defense response\"], [\"0009873\", \"ethylene-activated signaling pathway\"], [\"0009867\", \"jasmonic acid mediated signaling pathway\"], [\"2000022\", \"regulation of jasmonic acid mediated signaling pathway\"], [\"0009611\", \"response to wounding\"]]>]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Interaction_Network.show_annotated_objects"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Report is ready! Look at 'report.txt' file\n",
      "\n"
     ]
    }
   ],
   "source": [
    "##creating the final report with the networks and its annotations\n",
    "\n",
    "File.open(\"report.txt\", \"a\") { |f| f.write(\"\n",
    "### ========================================================================== REPORT ========================================================================== ###\n",
    "\n",
    "Analysis of interactions between the 168 co-expressed genes of the list. It has been applied filters for species and MIscore cutoff to the results of BAR database:\n",
    "    taxon ID for Arabidopsis thaliana = taxid:3702\n",
    "    cutoff = 0.485 (value for optimal score predictions found by calcuating the maximal Matthews correlation coefficient (MCC) in https://europepmc.org/article/MED/25652942)\n",
    "\n",
    "There are two kinds of interactions:\n",
    "    direct: between two or more genes of the list\n",
    "    indirect: two genes of the list has one or more interactors (not in the list) in common\n",
    "\n",
    "Although some genes of the list are present in more than one of the following networks (probably because they form a protein complex with many proteins of the list), I have created NETWORKS THAT HAVE ONLY 2 GENES OF THE LIST that interact with each other directly or/and indirectly through other interactors.\n",
    "\n",
    "\n",
    "### ========================================================================= NETWORKS ========================================================================= ###\n",
    "\") }\n",
    "\n",
    "\n",
    "\n",
    "for i in 0..Interaction_Network.show_annotated_objects.length-1\n",
    "  if Interaction_Network.show_annotated_objects[i].network_members.interactors == \"direct\"\n",
    "    File.open(\"report.txt\", \"a\") { |f| f.write(\"\\n\\nNETWORK ##{i+1}\") }\n",
    "    File.open(\"report.txt\", \"a\") { |f| f.write(\"\\nGene #{Interaction_Network.show_annotated_objects[i].network_members.agi_locus1} interacts directly with #{Interaction_Network.show_annotated_objects[i].network_members.agi_locus2}\")}\n",
    "    Interaction_Network.show_annotated_objects[i].kegg.each do |k|\n",
    "      File.open(\"report.txt\", \"a\") { |f| f.write(\"\\n\\tKEGG ID: #{k[0]}  Pathway Name: #{k[1]}\") }\n",
    "    end\n",
    "    Interaction_Network.show_annotated_objects[i].go.each do |g|\n",
    "      File.open(\"report.txt\", \"a\") { |f| f.write(\"\\n\\tGO ID: #{g[0]} GO Term: #{g[1]}\") }\n",
    "    end\n",
    "  else\n",
    "    File.open(\"report.txt\", \"a\") { |f| f.write(\"\\n\\nNETWORK ##{i+1}\") }\n",
    "    Interaction_Network.show_annotated_objects[i].network_members.interactors.each do |intn|\n",
    "      if intn.class == Array\n",
    "        intn.each do |intt|\n",
    "          if Interaction_Network.show_annotated_objects[i].network_members.interactors[0].class == String\n",
    "            File.open(\"report.txt\", \"a\") { |f| f.write(\"\\nGene #{Interaction_Network.show_annotated_objects[i].network_members.agi_locus1} interacts indirectly with #{Interaction_Network.show_annotated_objects[i].network_members.agi_locus2} by #{Interaction_Network.show_annotated_objects[i].network_members.interactors[0]}\") }\n",
    "            Interaction_Network.show_annotated_objects[i].kegg.each do |k|\n",
    "              File.open(\"report.txt\", \"a\") { |f| f.write(\"\\n\\tKEGG ID: #{k[0]}  Pathway Name: #{k[1]}\") }\n",
    "            end\n",
    "            Interaction_Network.show_annotated_objects[i].go.each do |g|\n",
    "              File.open(\"report.txt\", \"a\") { |f| f.write(\"\\n\\tGO ID: #{g[0]} GO Term: #{g[1]}\") }\n",
    "            end\n",
    "          else\n",
    "            File.open(\"report.txt\", \"a\") { |f| f.write(\"\\nGene #{Interaction_Network.show_annotated_objects[i].network_members.agi_locus1} interacts indirectly with #{Interaction_Network.show_annotated_objects[i].network_members.agi_locus2} by #{Interaction_Network.show_annotated_objects[i].network_members.interactors[0].join(\", \")}\") }\n",
    "          end\n",
    "          break\n",
    "        end\n",
    "      elsif intn == \"direct\"\n",
    "        File.open(\"report.txt\", \"a\") { |f| f.write(\"\\nGene #{Interaction_Network.show_annotated_objects[i].network_members.agi_locus1} interacts indirectly with #{Interaction_Network.show_annotated_objects[i].network_members.agi_locus2} by #{Interaction_Network.show_annotated_objects[i].network_members.interactors[0]}\") }\n",
    "        Interaction_Network.show_annotated_objects[i].kegg.each do |k|\n",
    "          File.open(\"report.txt\", \"a\") { |f| f.write(\"\\n\\tKEGG ID: #{k[0]}  Pathway Name: #{k[1]}\") }\n",
    "        end\n",
    "        Interaction_Network.show_annotated_objects[i].go.each do |g|\n",
    "          File.open(\"report.txt\", \"a\") { |f| f.write(\"\\n\\tGO ID: #{g[0]} GO Term: #{g[1]}\") }\n",
    "        end\n",
    "        break\n",
    "      else\n",
    "        File.open(\"report.txt\", \"a\") { |f| f.write(\"\\nGene #{Interaction_Network.show_annotated_objects[i].network_members.agi_locus1} interacts indirectly with #{Interaction_Network.show_annotated_objects[i].network_members.agi_locus2} by #{Interaction_Network.show_annotated_objects[i].network_members.interactors.join(\", \")}\") }\n",
    "        Interaction_Network.show_annotated_objects[i].kegg.each do |k|\n",
    "          File.open(\"report.txt\", \"a\") { |f| f.write(\"\\n\\tKEGG ID: #{k[0]}  Pathway Name: #{k[1]}\") }\n",
    "        end\n",
    "        Interaction_Network.show_annotated_objects[i].go.each do |g|\n",
    "          File.open(\"report.txt\", \"a\") { |f| f.write(\"\\n\\tGO ID: #{g[0]} GO Term: #{g[1]}\") }\n",
    "        end\n",
    "        break\n",
    "      end\n",
    "    end\n",
    "  end\n",
    "end\n",
    "\n",
    "puts \"\\nReport is ready! Look at 'report.txt' file\\n\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Ruby 2.5.1",
   "language": "ruby",
   "name": "ruby"
  },
  "language_info": {
   "file_extension": ".rb",
   "mimetype": "application/x-ruby",
   "name": "ruby",
   "version": "2.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
